

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>How to use the AuReMe workspace (default workflow) &mdash; aureme  documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="FAQ" href="faq.html" />
    <link rel="prev" title="How to use Docker?" href="docker.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> aureme
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="docker.html">How to use Docker?</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">How to use the AuReMe workspace (default workflow)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#requirements">Requirements</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#start-a-new-study">Start a new study</a></li>
<li class="toctree-l3"><a class="reference internal" href="#define-the-reference-database">Define the reference database</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#the-default-workflow">The default workflow</a></li>
<li class="toctree-l2"><a class="reference internal" href="#data-organization">Data organization</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#bridge-structure">Bridge structure</a></li>
<li class="toctree-l3"><a class="reference internal" href="#provide-input-files">Provide input files</a></li>
<li class="toctree-l3"><a class="reference internal" href="#image2check-input-files-validity">|image2|Check input files validity</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#orthology-based-reconstruction">Orthology-based reconstruction</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#method-pantograph">Method: Pantograph</a></li>
<li class="toctree-l3"><a class="reference internal" href="#orthology-based-inputs">Orthology-based inputs</a></li>
<li class="toctree-l3"><a class="reference internal" href="#orthology-based-run">Orthology-based run</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#annotation-based-reconstruction">Annotation-based reconstruction</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#method-pathway-tools">Method: Pathway Tools</a></li>
<li class="toctree-l3"><a class="reference internal" href="#annotation-based-inputs">Annotation-based inputs</a></li>
<li class="toctree-l3"><a class="reference internal" href="#annotation-based-run">Annotation-based run</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#merge-metabolic-networks">Merge metabolic networks</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gap-filling">Gap-filling</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#method-meneco">Method: Meneco</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#input">Input</a></li>
<li class="toctree-l4"><a class="reference internal" href="#run">Run</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#manual-curation">Manual curation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#add-a-reaction-from-the-database-or-delete-a-reaction-in-a-network">Add a reaction from the database or delete a reaction in a network</a></li>
<li class="toctree-l3"><a class="reference internal" href="#create-new-reaction-s-to-add-in-a-network">Create new reaction(s) to add in a network</a></li>
<li class="toctree-l3"><a class="reference internal" href="#apply-changes">Apply changes</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="faq.html">FAQ</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">aureme</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>How to use the AuReMe workspace (default workflow)</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/aureme.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <p>AuReMe documentation</p>
<div class="section" id="how-to-use-the-aureme-workspace-default-workflow">
<h1>How to use the AuReMe workspace (default workflow)<a class="headerlink" href="#how-to-use-the-aureme-workspace-default-workflow" title="Permalink to this headline">¶</a></h1>
<div class="section" id="requirements">
<h2>Requirements<a class="headerlink" href="#requirements" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li>Create your Docker container as explained in the previous step
<strong>“Running a Docker container”</strong>, start the container and go inside.</li>
</ol>
<div class="section" id="start-a-new-study">
<h3>Start a new study<a class="headerlink" href="#start-a-new-study" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple" start="2">
<li>Use the following command to start a new study. Choose an identifier
for this study (ex: replace <strong>*test*</strong> by your organism name). In
order to illustrate this documentation, we will use <strong>*test*</strong> as a
run identifier.</li>
</ol>
<blockquote>
<div><p>Now you will find on your own computer (host), in your <strong>*bridge*</strong>
directory, a folder <strong>*test*</strong> with many subdirectory and files.
This is your work directory, on which AuReMe is going to run.</p>
<p>Notice that from now until the end of the process, every command
will be stored as a log in the <strong>*bridgetestlog.txt*</strong> file. The
whole output of these commands will also be stored in the
<strong>*bridgetestfull_log.txt*</strong> file.</p>
<p>If you wish NOT to store such logs, you can use the <strong>quiet</strong>
argument in your command(s). This will redirect the output on the
terminal</p>
<p>For example:</p>
<p>For further details on the log files, please see the <strong>“FAQ How to
manage the log files”</strong> chapter.</p>
</div></blockquote>
<ol class="arabic simple" start="3">
<li>To get an overview of AuReMe, you can get a sample by using this
command.</li>
</ol>
</div>
<div class="section" id="define-the-reference-database">
<h3>Define the reference database<a class="headerlink" href="#define-the-reference-database" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple" start="4">
<li>The final step is to define which reference database to use. The
available databases are listed in your terminal when you create a new
study. If needed, use this command to display them again.</li>
</ol>
<blockquote>
<div>This reference database is needed to:</div></blockquote>
<ul class="simple">
<li><dl class="first docutils">
<dt>Be able to match all the identifiers of the entities of metabolic</dt>
<dd>networks</dd>
</dl>
</li>
<li>Gap-fill the metabolic network in the gap-filling step</li>
<li>Uniforms the data in one unique database</li>
</ul>
<blockquote>
<div><p>To select one, replace the corresponding path in the configuration
file: <strong>*config.txt*</strong>, in the <strong>*DATA_BASE*</strong> variable. Or comment
the line if you don’t want/can’t use a database.</p>
<p>The <strong>*config.txt*</strong> file is stored at the root of your <strong>*test*</strong>
folder.</p>
</div></blockquote>
</div>
</div>
<div class="section" id="the-default-workflow">
<h2>The default workflow<a class="headerlink" href="#the-default-workflow" title="Permalink to this headline">¶</a></h2>
<p>By default, the AuReMe workspace includes an automatic workflow for
metabolic network reconstruction. This workflow runs several
pre-installed tools and generates diverse output files. The process can
be either run entirely in a single command, or run step by step to
personalize it or do some intermediary analysis.</p>
<p><a href="#id1"><span class="problematic" id="id2">|image0|</span></a>For instance, if you run the <strong>*draft*</strong> command (see <strong>“Merge
metabolic networks”</strong>), it will run all the previous steps automatically
as described in the following figure. This figure details the steps of
the default workflow.</p>
</div>
<div class="section" id="data-organization">
<h2>Data organization<a class="headerlink" href="#data-organization" title="Permalink to this headline">¶</a></h2>
<div class="section" id="bridge-structure">
<h3>Bridge structure<a class="headerlink" href="#bridge-structure" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><p>The <strong>*bridge*</strong> directory will store all your input data you will
provide, and all the result files the workflow is going to create.</p>
<p><strong>*analysis*</strong>: all output files of the analysis processes</p>
<p><strong>*annotation_based_reconstruction*</strong>: if you want to use annotated
genomes (to run the annotation-based reconstruction part of the
workflow), put here all the output files of the annotation tool. For
instance, with Pathway Tools, copy-paste the whole PGDB directory
(see below “Annotation-based reconstruction” for more details).</p>
<p><strong>*database*</strong>: if you want to use your own database put in this
folder your database in padmet format, if you have a sbml convert
this file to padmet (see <strong>“FAQ How to convert files to different
formats”</strong>)</p>
<p><strong>*gapfilling/original_output*</strong>: if you run the metabolic network
reconstruction with gap-filling, will contain all the output files
of gap-filling tools before any post-process from AuReMe.</p>
<p><a href="#id3"><span class="problematic" id="id4">|image1|</span></a><strong>*genomic-data*</strong>: the directory in which to put the
genomic data on your studied organism, that is to say either a
Genbank (.gbk) or a proteome (.faa).</p>
<p><strong>*growth_medium*</strong>: description of the set of metabolites that is
available to initiate the metabolism (growth medium), that is to say
the seed compounds (.txt) (see <strong>“FAQ How to manage growth
medium?”</strong>).</p>
<p><strong>*manual_curation*</strong>: all the file to describe the manual curation
you want to apply on your metabolic network (either adding, deleting
or modifying reactions).</p>
<p><strong>*networks*</strong>: all the metabolic networks used or created during
the reconstruction process</p>
<p><strong>*networks* *external_network*</strong>: put here all existing metabolic
networks (.sbml) you want to use. Enables to merge them with the
ones created thanks to other methods.</p>
<p><strong>*networks* *output_annotation_based_reconstruction*</strong>: will
contain the processed network from the annotation based
reconstruction, after the pre-processing of the data from the
<strong>*annotation_based_reconstruction*</strong> directory (if you filled this
one).</p>
<p><strong>*networks* *output_orthology_based_reconstruction*</strong>: will contain
the processed network from the orthology based reconstruction, after
the pre-processing of the data from the
<strong>*orthology_based_reconstruction*</strong> directory (if you have run this
part of the workflow).</p>
<p><strong>*orthology_based_reconstruction*</strong>: if you want to use model
organisms (to run orthology-based reconstruction part of the
workflow), put here the proteome (.faa or .gbk) and the metabolic
network (.sbml) of your model (see below “Orthology-based
reconstruction” for more details).</p>
<p><strong>*targets_compounds*</strong>: description of the set target compounds
(.txt), that is to say metabolites whose production is supposed to
be achieved by the metabolism of the species under study (components
of the biomass reactions or other metabolites).</p>
</div></blockquote>
</div>
<div class="section" id="provide-input-files">
<h3>Provide input files<a class="headerlink" href="#provide-input-files" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><p>First of all, you have to provide to AuReMe all the input files
needed for the different steps you want to run in the workflow. The
steps can be optional or run several times, at different phases of
the process. However, you have to store the input data for each
steps, observing the architecture described above for the
<strong>*bridge*</strong> directory (see <strong>“Data organization Bridge
structure”</strong>).</p>
<p>Here is the list of input you have to provide to run the pre-set
default workflow. If you want to run only part of it, please see the
corresponding sections and the chapter <strong>“How to create your
‘à-la-carte’ workflow”</strong>.</p>
</div></blockquote>
<ul class="simple">
<li><strong>See “Orthology-based reconstruction Inputs”</strong></li>
<li><strong>See “Annotation-based reconstruction Inputs”</strong></li>
<li><strong>External source for reconstruction</strong></li>
</ul>
<blockquote>
<div>If you already have one or several external metabolic networks for
your studied species and you want to improve them, just copy-paste
them (SBML format) in the <strong>*networksexternal_network*</strong> folder.</div></blockquote>
</div>
<div class="section" id="image2check-input-files-validity">
<h3><a href="#id5"><span class="problematic" id="id6">|image2|</span></a>Check input files validity<a class="headerlink" href="#image2check-input-files-validity" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple">
<li>IMPORTANT: Always check the validity of the inputs before running any
workflow task, and after having put every input files needed for the
steps of the workflow. This will verify the format and consistency of
your data for a better quality result. Moreover, it will generate all
the supplementary files needed by the workflow tools and put them
into the corresponding folders. For more information about input
files validity see <strong>“FAQ What is checked in my input files?”</strong>.</li>
</ol>
<blockquote>
<div>For this purpose, use this command:</div></blockquote>
</div>
</div>
<div class="section" id="orthology-based-reconstruction">
<h2>Orthology-based reconstruction<a class="headerlink" href="#orthology-based-reconstruction" title="Permalink to this headline">¶</a></h2>
<div class="section" id="method-pantograph">
<h3>Method: Pantograph<a class="headerlink" href="#method-pantograph" title="Permalink to this headline">¶</a></h3>
<p>Input files:</p>
<p><a href="#id7"><span class="problematic" id="id8">|image3|</span></a>- Required for the orthology-based reconstruction (method:
Pantograph):</p>
<ul class="simple">
<li>Genbank or Proteome of your studied organism (.gbk or .faa)</li>
<li>Genbank or Proteome of your reference organism (.gbk or .faa)</li>
<li>Metabolic network of your reference organism (.sbml)</li>
<li>(option) a dictionary file if genes ids used in metabolic network are</li>
</ul>
<p>different with gbk/faa (.txt)</p>
<p>Result file:</p>
<p>/test</p>
<p>|–orthology_based_reconstruction</p>
<p>| |– <em>model_a</em></p>
<p>| |– <strong>original_output_pantograph_*model_a*.sbml</strong></p>
<p>|– networks</p>
<p>|– orthology_based_reconstruction</p>
<p>|– pantograph</p>
<p>|– <strong>output_pantograph_</strong><em>model_a</em><strong>.sbml</strong></p>
</div>
<div class="section" id="orthology-based-inputs">
<h3>Orthology-based inputs<a class="headerlink" href="#orthology-based-inputs" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple">
<li>Put all the available genomic data of the studied organism in the
folder <strong>*genomic_data*</strong>, either a Genbank (.gbk) or a Fasta (.faa)
file. WARNING: give them these exact names (respectively):
GBK_study.gbk and FAA_study.faa.</li>
<li>For each reference organism you want to use, create a folder in the
folder <strong>*orthology_based_reconstruction*</strong>. Give it the name of your
model organism (e.g. <strong>*model_a*</strong>).</li>
</ol>
<blockquote>
<div>On a Linux operating system, here is the command to create a new
folder named <strong>*model_a*</strong>:</div></blockquote>
<ol class="arabic simple" start="3">
<li>In each folder, put:<ul>
<li><dl class="first docutils">
<dt>the Genbank file of your model organism, with the exact name</dt>
<dd>GBK_model.gbk</dd>
</dl>
</li>
</ul>
</li>
</ol>
<blockquote>
<div>OR the proteome of your model organism, with the exact name
FAA_model.faa</div></blockquote>
<ul class="simple">
<li><dl class="first docutils">
<dt>the metabolic network of your model organism, with the exact name</dt>
<dd>metabolic_model.sbml</dd>
</dl>
</li>
</ul>
<ol class="arabic" start="4">
<li><div class="first line-block">
<div class="line">The genome (or proteome) and the metabolic network of your model
organism have to contain the same kind genes (or proteins)
identifiers to be comparable. If not enough genes (or proteins) are
in common between the two files, the process will stop to avoid
poor quality data production.</div>
<div class="line">If you want to pursue on the process, please provide a dictionary
file between the gene (or protein) identifiers of these two files.
Name this dictionary <strong>*dict_genes.txt*</strong>. Here is the dictionary
file format asked (tabulation separated values):</div>
</div>
</li>
</ol>
</div>
<div class="section" id="orthology-based-run">
<h3>Orthology-based run<a class="headerlink" href="#orthology-based-run" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple" start="5">
<li>Important: Remember to check the validity of the inputs before
running any workflow task. If you want to run only the
orthology-based reconstruction, use now this command:</li>
<li>To run only the orthology-based reconstruction, use this command:</li>
<li>IMPORTANT: Because the metabolic network from the reference organism
could came from different databases, it’s critical to check the
database of each network and if needed convert the network to your
reference database selected (see <strong>“How to use the AuReMe workspace
(default workflow) Define the reference database”</strong>).</li>
</ol>
<blockquote>
<div>The previous command will check the database of the file
output_pantograph_mode_a.sbml, if the database is different for the
reference, use the next command to create a mapping file to metacyc
database. For more information about sbml mapping see <strong>“FAQ How to
map a sbml to another database?”</strong>.</div></blockquote>
</div>
</div>
<div class="section" id="annotation-based-reconstruction">
<h2>Annotation-based reconstruction<a class="headerlink" href="#annotation-based-reconstruction" title="Permalink to this headline">¶</a></h2>
<div class="section" id="method-pathway-tools">
<h3>Method: Pathway Tools<a class="headerlink" href="#method-pathway-tools" title="Permalink to this headline">¶</a></h3>
<p><a href="#id9"><span class="problematic" id="id10">|image4|</span></a></p>
<p>Input files:</p>
<ul class="simple">
<li>Required for the annotation-based reconstruction (method: Pathway</li>
</ul>
<p>Tools):</p>
<p>The output of Pathway tools (PGDB folder)</p>
<p>Result file:</p>
<p>/test</p>
<p>|– networks</p>
<p>|– annotation_based_reconstruction</p>
<p>|– pathwaytools</p>
<p>|– <strong>output_pathwaytools_</strong><em>genome_a</em><strong>.padmet</strong></p>
<p>|– <strong>output_pathwaytools_</strong><em>genome_b</em><strong>.padmet</strong></p>
</div>
<div class="section" id="annotation-based-inputs">
<span id="inputs-1"></span><h3>Annotation-based inputs<a class="headerlink" href="#annotation-based-inputs" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple">
<li>Put the output of Pathway Tools (the whole PGDB directory) in the
folder <strong>*annotation_based_reconstruction*</strong></li>
<li>If you have run several times Pathway Tools and want to use all of
these annotations, just copy-paste the other PGDB folders in the
<strong>*annotation_based_reconstruction*</strong> directory.</li>
</ol>
</div>
<div class="section" id="annotation-based-run">
<span id="run-1"></span><h3>Annotation-based run<a class="headerlink" href="#annotation-based-run" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple" start="3">
<li>Important: Remember to check the validity of the inputs before
running any workflow task. If you want to run only the
annotation-based reconstruction, use now this command:</li>
<li>To run only the annotation-based reconstruction, use this command.</li>
</ol>
</div>
</div>
<div class="section" id="merge-metabolic-networks">
<h2>Merge metabolic networks<a class="headerlink" href="#merge-metabolic-networks" title="Permalink to this headline">¶</a></h2>
<p><a href="#id11"><span class="problematic" id="id12">|image5|</span></a></p>
<p>Input files:</p>
<ul class="simple">
<li>metabolic networks in the <strong>*networks*</strong> directory</li>
</ul>
<p>Result files:</p>
<p>/test</p>
<p>|– netowrks</p>
<p>|– <strong>draft.padmet</strong></p>
<p>To merge all available networks from the <strong>*networks*</strong> directory into
one metabolic network, merging all data on the studied species, run this
command:</p>
<p>Note that you can also add external metabolic network to create the
draft (see <strong>“Data organization Provide input files External source for
reconstruction”</strong>).</p>
<p>IMPORTANT: Before merging your networks, check if not already done if
all the sbml are using the reference database. Also check the
compartment ids used in each of them, delete and change compartment if
need.</p>
<p>For example: if a sbml is using KEGG database but your reference
database is metacyc, you will have to map this sbml to create a mapping
file which will be used automatically in the merging process.</p>
<p>If a sbml contains a compartment id like ‘C_c’ and another contains ‘c’,
although they correspond to the same compartment ‘cytosol’ because of
different ids, a compound in ‘C_c’ is not the same as a compound in ‘c’,
therefore there will be a loss of connectivity in the network. see
<strong>“FAQ How to map a sbml to another database?” and “FAQ How to manage
compartment?”</strong></p>
</div>
<div class="section" id="gap-filling">
<h2>Gap-filling<a class="headerlink" href="#gap-filling" title="Permalink to this headline">¶</a></h2>
<div class="section" id="method-meneco">
<h3>Method: Meneco<a class="headerlink" href="#method-meneco" title="Permalink to this headline">¶</a></h3>
<p><a href="#id13"><span class="problematic" id="id14">|image6|</span></a>Input files:</p>
<ul class="simple">
<li>Required for the gap-filling (method: Meneco):</li>
<li>A metabolic network reference database (.padmet or .sbml)</li>
</ul>
<p>(metacyc 20.5, 22.0, BIGG and ModelSeed are available by default)</p>
<ul class="simple">
<li>Seed and target metabolites (.txt)</li>
<li>A metabolic network to fill (typically created during the previous</li>
</ul>
<p>steps)</p>
<p>Result files:</p>
<p>/test</p>
<p>|– netowrks</p>
<p>|– <em>network_name</em><strong>.sbml</strong></p>
<p>|– <em>network_name</em><strong>.padmet</strong></p>
<p>|– gapfilling</p>
<p>|– original_output</p>
<p>| |– <strong>meneco_output_</strong> <em>network_name</em><strong>.txt</strong></p>
<p>|– <strong>gapfilling_solution_</strong> <em>network_name</em><strong>.csv</strong></p>
<div class="section" id="input">
<h4>Input<a class="headerlink" href="#input" title="Permalink to this headline">¶</a></h4>
<ol class="arabic simple">
<li>You must have selected a reference database to fill-in the potential
gaps in the metabolic network. If it is not done yet, please see
<strong>“Requirements Define the reference database ”</strong></li>
<li>Put the seeds file (named seeds.txt) in the <strong>*growth_medium*</strong>
folder. The seed compounds are the description of the set of
metabolites that is available to initiate the metabolism (growth
medium).</li>
</ol>
<blockquote>
<div>Here is as example of the seed file format:</div></blockquote>
<ol class="arabic simple" start="3">
<li>Set the growth medium using this command:</li>
</ol>
<blockquote>
<div><p>For more details on the medium settings, see <strong>“FAQ How to manage
growth medium?”</strong>.</p>
<p>WARNING: If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be overwritten.</p>
</div></blockquote>
<ol class="arabic simple" start="4">
<li>Put the target file (named targets.txt) in the
<strong>*targets_compounds*</strong> folder. The targets are metabolites whose
production is supposed to be achieved by the metabolism of the
species under study (components of the biomass reactions or other
metabolites).</li>
</ol>
<blockquote>
<div>Here is as example of the seed file format:</div></blockquote>
<ol class="arabic simple" start="5">
<li>You will have to indicate which metabolic network you want to
gap-fill with the Meneco software. If you want to gap-fill a network
created in the previous steps, there is nothing to do. Otherwise, put
the network you want to gap-fill (PADMET format) in the
<strong>*networks*</strong> directory.</li>
</ol>
</div>
<div class="section" id="run">
<span id="run-2"></span><h4>Run<a class="headerlink" href="#run" title="Permalink to this headline">¶</a></h4>
<ol class="arabic simple" start="6">
<li>(optional step) To generate the gap-filling solution run this
command:</li>
</ol>
<blockquote>
<div><p>Note: Do not forget the quotation marks.</p>
<p>It will calculate the gap-filling solution on the <em>network_name</em>
network (in the <strong>*networks*</strong> directory) and put it into the
<strong>*gapfilling*</strong> directory as gapfilling_solution_network_name.csv</p>
</div></blockquote>
<ol class="arabic simple" start="7">
<li>To generate the gap-filled network (and run step 6), run this
command:</li>
</ol>
<blockquote>
<div><p>Note: Do not forget the quotation marks.</p>
<p>It will calculate the gap-filling solution (if it is not yet done)
on the <em>network_name</em> network (in the <strong>*networks*</strong> directory) and
put it into the <strong>*gapfilling*</strong> directory. Then it will generate
the metabolic network (<em>new_network_name</em>), completed with the
gap-filling solution, in the <strong>*networks*</strong> directory.</p>
</div></blockquote>
<p>Note that you can first generate the solution, modify it, then generate
the gap-filled network.</p>
<blockquote>
<div>WARNING: If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be overwritten.</div></blockquote>
</div>
</div>
</div>
<div class="section" id="manual-curation">
<h2>Manual curation<a class="headerlink" href="#manual-curation" title="Permalink to this headline">¶</a></h2>
<p>This step can be done several times and at any moment of the workflow.</p>
<ol class="arabic simple">
<li>Describe the manual curation(s) you want to apply by filling the
corresponding form(s) as explained below.</li>
</ol>
<p>Important note: It is highly recommanded to create a new form file
(.csv) each time you want to apply other changes, in order to keep
tracks of them.</p>
<div class="section" id="add-a-reaction-from-the-database-or-delete-a-reaction-in-a-network">
<h3>Add a reaction from the database or delete a reaction in a network<a class="headerlink" href="#add-a-reaction-from-the-database-or-delete-a-reaction-in-a-network" title="Permalink to this headline">¶</a></h3>
<ol class="loweralpha simple">
<li>Copy from the folder <strong>manual_curationdatatemplate</strong> the file
<strong>reaction_to_add_delete.csv</strong> and paste it into the
<strong>manual_curationdata</strong> directory (this way on Linux operating
systems):</li>
<li>Fill this file (follow the exemple in the template).</li>
</ol>
</div>
<div class="section" id="create-new-reaction-s-to-add-in-a-network">
<h3>Create new reaction(s) to add in a network<a class="headerlink" href="#create-new-reaction-s-to-add-in-a-network" title="Permalink to this headline">¶</a></h3>
<ol class="loweralpha simple">
<li>Copy from the folder <strong>manual_curationdatatemplate</strong> the file
<strong>*reaction_creator.csv*</strong> and paste it into the
<strong>manual_curationdata</strong> directory (this way on Linux operating
systems):</li>
<li>Fill this file (follow the exemple in the template).</li>
</ol>
</div>
<div class="section" id="apply-changes">
<h3>Apply changes<a class="headerlink" href="#apply-changes" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple">
<li>To apply the changes described in the <em>my_form_file.csv</em> form file,
run this command:</li>
</ol>
<blockquote>
<div><p>WARNING: If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be overwritten.</p>
<p>Note that all the manual curations made are stored in history files
in the <strong>manual_curationhistory</strong> directory. You can use them to do
the same corrections on other networks for example.</p>
</div></blockquote>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="faq.html" class="btn btn-neutral float-right" title="FAQ" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="docker.html" class="btn btn-neutral float-left" title="How to use Docker?" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Meziane AITE

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>