

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Table des matières &mdash; aureme  documentation</title>
  

  
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Welcome to aureme’s documentation!" href="index.html" /> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> aureme
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Table des matières</a></li>
<li class="toctree-l1"><a class="reference internal" href="#how-to-use-docker">How to use Docker</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#running-a-docker-container">Running a Docker container</a></li>
<li class="toctree-l2"><a class="reference internal" href="#some-tips-about-docker">Some tips <strong>about</strong> Docker</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="#how-to-use-the-aureme-workspace-default-workflow">How to use the AuReMe workspace (default workflow)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#requirements">Requirements</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#start-a-new-study">Start a new study</a></li>
<li class="toctree-l3"><a class="reference internal" href="#define-the-reference-database">Define the reference database</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#the-default-workflow">The default workflow</a></li>
<li class="toctree-l2"><a class="reference internal" href="#data-organization">Data organization</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#bridge-structure">Bridge structure</a></li>
<li class="toctree-l3"><a class="reference internal" href="#provide-input-files">Provide input files</a></li>
<li class="toctree-l3"><a class="reference internal" href="#image2check-input-files-validity">Check input files validity</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#orthology-based-reconstruction">Orthology-based reconstruction</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#method-pantograph">Method: Pantograph</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#inputs">Inputs</a></li>
<li class="toctree-l4"><a class="reference internal" href="#run">Run</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#annotation-based-reconstruction">Annotation-based reconstruction</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#method-pathway-tools">Method: Pathway Tools</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#inputs-1">Inputs</a></li>
<li class="toctree-l4"><a class="reference internal" href="#run-1">Run</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#merge-metabolic-networks">Merge metabolic networks</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gap-filling">Gap-filling</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#method-meneco">Method: Meneco</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#input">Input</a></li>
<li class="toctree-l4"><a class="reference internal" href="#run-2">Run</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#manual-curation"><strong>Manual curation</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#add-a-reaction-from-the-database-or-delete-a-reaction-in-a-network">Add a reaction from the database or delete a reaction in a network</a></li>
<li class="toctree-l3"><a class="reference internal" href="#create-new-reaction-s-to-add-in-a-network">Create new reaction(s) to add in a network</a></li>
<li class="toctree-l3"><a class="reference internal" href="#apply-changes">Apply changes</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="#faq">FAQ</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#can-i-have-a-sample-of-aureme">Can I have a sample of AuReMe?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-convert-files-to-different-formats">How to convert files to different formats?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-manage-growth-medium">How to manage growth medium?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-manage-metabolic-network-compartment">How to manage metabolic network compartment?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-manage-the-log-files">How to manage the log files?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-reproduce-studies">How to reproduce studies?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-create-a-new-a-la-carte-workflow">How to create a new ‘à-la-carte’ workflow?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-choose-another-reference-database">How to choose another reference database?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-is-checked-in-my-input-files">What is checked in my input files?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-is-the-makefile">What is the Makefile?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-is-the-config-txt-file">What is the config.txt file?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-regenerate-a-new-database-version">How to regenerate a new database version?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-map-a-metabolic-network-on-another-database">How to map a metabolic network on another database?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-generate-reports-on-results">How to generate reports on results?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-generate-wiki">How to generate Wiki?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-connect-to-pathway-tools">How to connect to Pathway-tools?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-are-artefacts">What are “artefacts”?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-process-flux-balance-analysis">How to process Flux Balance Analysis?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-set-an-objective-reaction">How to set an objective reaction?</a></li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">aureme</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Table des matières</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/AuReMe_documentation_v9.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <p>The AuReMe Workspace</p>
<div class="section" id="table-des-matieres">
<h1>Table des matières<a class="headerlink" href="#table-des-matieres" title="Permalink to this headline">¶</a></h1>
<p><a class="reference external" href="#how-to-use-docker">How to use Docker 3</a></p>
<p><a class="reference external" href="#running-a-docker-container">Running a Docker container 3</a></p>
<p><a class="reference external" href="#some-tips-about-docker">Some tips about Docker 3</a></p>
<p><a class="reference external" href="#how-to-use-the-aureme-workspace-default-workflow">How to use the AuReMe workspace (default workflow)
4</a></p>
<p><a class="reference external" href="#requirements">Requirements 4</a></p>
<p><a class="reference external" href="#start-a-new-study">Start a new study 4</a></p>
<p><a class="reference external" href="#define-the-reference-database">Define the reference database 4</a></p>
<p><a class="reference external" href="#the-default-workflow">The default workflow 5</a></p>
<p><a class="reference external" href="#data-organization">Data organization 5</a></p>
<p><a class="reference external" href="#bridge-structure">Bridge structure 5</a></p>
<p><a class="reference external" href="#provide-input-files">Provide input files 6</a></p>
<p><a class="reference external" href="#check-input-files-validity">Check input files validity 7</a></p>
<p><a class="reference external" href="#orthology-based-reconstruction">Orthology-based reconstruction 7</a></p>
<p><a class="reference external" href="#method-pantograph">Method: Pantograph 7</a></p>
<p><a class="reference external" href="#annotation-based-reconstruction">Annotation-based reconstruction 8</a></p>
<p><a class="reference external" href="#method-pathway-tools">Method: Pathway Tools 8</a></p>
<p><a class="reference external" href="#merge-metabolic-networks">Merge metabolic networks 9</a></p>
<p><a class="reference external" href="#gap-filling">Gap-filling 9</a></p>
<p><a class="reference external" href="#method-meneco">Method: Meneco 9</a></p>
<p><a class="reference external" href="#manual-curation">Manual curation 10</a></p>
<p><a class="reference external" href="#add-a-reaction-from-the-database-or-delete-a-reaction-in-a-network">Add a reaction from the database or delete a reaction in a network
10</a></p>
<p><a class="reference external" href="#create-new-reactions-to-add-in-a-network">Create new reaction(s) to add in a network
11</a></p>
<p><a class="reference external" href="#apply-changes">Apply changes 11</a></p>
<p><a class="reference external" href="#faq">FAQ 12</a></p>
<p><a class="reference external" href="#can-i-have-a-sample-of-aureme">Can I have a sample of AuReMe? 12</a></p>
<p><a class="reference external" href="#how-to-convert-files-to-different-formats">How to convert files to different formats?
12</a></p>
<p><a class="reference external" href="#how-to-manage-growth-medium">How to manage growth medium? 12</a></p>
<p><a class="reference external" href="#how-to-manage-metabolic-network-compartment">How to manage metabolic network compartment?
12</a></p>
<p><a class="reference external" href="#how-to-manage-the-log-files">How to manage the log files? 13</a></p>
<p><a class="reference external" href="#how-to-reproduce-studies">How to reproduce studies? 13</a></p>
<p><a class="reference external" href="#how-to-create-a-new-à-la-carte-workflow">How to create a new ‘à-la-carte’ workflow?
13</a></p>
<p><a class="reference external" href="#how-to-choose-another-reference-database">How to choose another reference database?
14</a></p>
<p><a class="reference external" href="#what-is-checked-in-my-input-files">What is checked in my input files?
14</a></p>
<p><a class="reference external" href="#how-to-regenerate-a-new-database-version">How to regenerate a new database version?
14</a></p>
<p><a class="reference external" href="#how-to-map-a-metabolic-network-on-another-database">How to map a metabolic network on another database?
14</a></p>
<p><a class="reference external" href="#how-to-generate-reports-on-results">How to generate reports on results?
15</a></p>
<p><a class="reference external" href="#how-to-generate-wiki">How to generate Wiki? 15</a></p>
<p><a class="reference external" href="#how-to-connect-to-pathway-tools">How to connect to Pathway-tools?
16</a></p>
<p><a class="reference external" href="#_Toc527496019">The Docker container 16</a></p>
<p><a class="reference external" href="#what-are-artefacts">What are “artefacts”? 17</a></p>
<p><a class="reference external" href="#how-to-process-flux-balance-analysis">How to process Flux Balance Analysis?
17</a></p>
<p><a class="reference external" href="#how-to-set-an-objective-reaction">How to set an objective reaction?
17</a></p>
</div>
<div class="section" id="how-to-use-docker">
<h1>How to use Docker<a class="headerlink" href="#how-to-use-docker" title="Permalink to this headline">¶</a></h1>
<p>Requirements: Docker (v 1.10 at least)</p>
<p>To install Docker, please follow the instructions on
<a class="reference external" href="https://www.docker.com/">docker.com</a>, considering your operating
system*.</p>
<p>*On Mac OS: requires at least Yosemite 10.10.4</p>
<p>*On Windows: requires at least Windows 8</p>
<div class="section" id="running-a-docker-container">
<h2>Running a Docker container<a class="headerlink" href="#running-a-docker-container" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li><dl class="first docutils">
<dt>Launch the Docker machine (see the instruction on</dt>
<dd><a class="reference external" href="https://www.docker.com/">docker.com</a>). For example:</dd>
</dl>
</li>
</ol>
<ul class="simple">
<li>On Fedora: <strong>sudo systemctl start docker</strong></li>
<li>On Mac OS and Windows: run the Docker launcher</li>
</ul>
<ol class="arabic simple" start="2">
<li>Download the AuReMe Docker image</li>
<li><dl class="first docutils">
<dt>To verify that the image has been downloaded correctly, check it in</dt>
<dd>the list of your local images:</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Create a folder that will serve as a&nbsp;bridge to share data from/to the</dt>
<dd>Docker container. Let us call it <strong>*bridge*</strong> for instance.</dd>
</dl>
</li>
<li>Create a Docker container from the following image with this command:</li>
</ol>
<p>The path given for –v is the one to the shared directory on your host
machine</p>
<p><strong>This path has to end on the directory name</strong> (without any <strong>/</strong> at the
end)</p>
<p><strong>The path has to be complete</strong> (from <strong>/users</strong> or from <a href="#id1"><span class="problematic" id="id2">**</span></a>C:\** for
Windows users)</p>
<p>After the ‘:’ is the name of the mirror directory in the Docker
container. Please do not change it.</p>
<p>For Windows users, be careful, you have to indicate your path this way:</p>
<p>You can create as many container as you wish, as long as you give them
different names.</p>
<p>Your AuReMe container is now running and correctly installed.
Congratulations! You are now inside the container <strong>*aureme-cont*</strong>.</p>
</div>
<div class="section" id="some-tips-about-docker">
<h2>Some tips <strong>about</strong> Docker<a class="headerlink" href="#some-tips-about-docker" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>To exit the container, tape <strong>exit.</strong></li>
<li>To list all your containers:</li>
</ul>
<p>Remark that you can see, with this command, the state of your containers
in the STATUS column: <strong>up</strong> (running, you can connect to it),
<strong>exited</strong> (stopped, need to be started again)</p>
<ul class="simple">
<li>To start or stop the container (from your host):</li>
<li>If you want to go inside a started/running container:</li>
<li>To delete a container: <strong>docker rm *container_id* (</strong>or <strong>*name*)</strong></li>
<li>To delete an image: <strong>docker rmi *image_id* (</strong>or <strong>*name*)</strong></li>
</ul>
<p>This is impossible if any existing container has been created from it.
Delete all dependent containers first.</p>
</div>
</div>
<div class="section" id="how-to-use-the-aureme-workspace-default-workflow">
<h1>How to use the AuReMe workspace (default workflow)<a class="headerlink" href="#how-to-use-the-aureme-workspace-default-workflow" title="Permalink to this headline">¶</a></h1>
<div class="section" id="requirements">
<h2>Requirements<a class="headerlink" href="#requirements" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li>Create your Docker container as explained in the previous step
<strong>“Running a Docker container”</strong>, start the container and go inside.</li>
</ol>
<div class="section" id="start-a-new-study">
<h3>Start a new study<a class="headerlink" href="#start-a-new-study" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple" start="2">
<li>Use the following command to start a new study. Choose an identifier
for this study (ex: replace <strong>*test*</strong> by your organism name). In
order to illustrate this documentation, we will use <strong>*test*</strong> as a
run identifier.</li>
</ol>
<blockquote>
<div><p>Now you will find on your own computer (host), in your <strong>*bridge*</strong>
directory, a folder <strong>*test*</strong> with many subdirectory and files.
This is your work directory, on which AuReMe is going to run.</p>
<p>Notice that from now until the end of the process, every command
will be stored as a log in the <strong>*bridgetestlog.txt*</strong> file. The
whole output of these commands will also be stored in the
<strong>*bridgetestfull_log.txt*</strong> file.</p>
<p>If you wish NOT to store such logs, you can use the <strong>quiet</strong>
argument in your command(s). This will redirect the output on the
terminal</p>
<p>For example:</p>
<p>For further details on the log files, please see the <strong>“FAQ How to
manage the log files”</strong> chapter.</p>
</div></blockquote>
<ol class="arabic simple" start="3">
<li>To get an overview of AuReMe, you can get a sample by using this
command.</li>
</ol>
</div>
<div class="section" id="define-the-reference-database">
<h3>Define the reference database<a class="headerlink" href="#define-the-reference-database" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple" start="4">
<li>The final step is to define which reference database to use. The
available databases are listed in your terminal when you create a new
study. If needed, use this command to display them again.</li>
</ol>
<blockquote>
<div>This reference database is needed to:</div></blockquote>
<ul class="simple">
<li><dl class="first docutils">
<dt>Be able to match all the identifiers of the entities of metabolic</dt>
<dd>networks</dd>
</dl>
</li>
<li>Gap-fill the metabolic network in the gap-filling step</li>
<li>Uniforms the data in one unique database</li>
</ul>
<blockquote>
<div><p>To select one, replace the corresponding path in the configuration
file: <strong>*config.txt*</strong>, in the <strong>*DATA_BASE*</strong> variable. Or comment
the line if you don’t want/can’t use a database.</p>
<p>The <strong>*config.txt*</strong> file is stored at the root of your <strong>*test*</strong>
folder.</p>
</div></blockquote>
</div>
</div>
<div class="section" id="the-default-workflow">
<h2>The default workflow<a class="headerlink" href="#the-default-workflow" title="Permalink to this headline">¶</a></h2>
<p>By default, the AuReMe workspace includes an automatic workflow for
metabolic network reconstruction. This workflow runs several
pre-installed tools and generates diverse output files. The process can
be either run entirely in a single command, or run step by step to
personalize it or do some intermediary analysis.</p>
<p><a class="reference internal" href="media/image1.png"><img alt="image0" src="media/image1.png" style="width: 4.10069in; height: 4.27986in;" /></a>For instance, if you run the <strong>*draft*</strong> command (see <strong>“Merge
metabolic networks”</strong>), it will run all the previous steps automatically
as described in the following figure. This figure details the steps of
the default workflow.</p>
</div>
<div class="section" id="data-organization">
<h2>Data organization<a class="headerlink" href="#data-organization" title="Permalink to this headline">¶</a></h2>
<div class="section" id="bridge-structure">
<h3>Bridge structure<a class="headerlink" href="#bridge-structure" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><p>The <strong>*bridge*</strong> directory will store all your input data you will
provide, and all the result files the workflow is going to create.</p>
<p><strong>*analysis*</strong>: all output files of the analysis processes</p>
<p><strong>*annotation_based_reconstruction*</strong>: if you want to use annotated
genomes (to run the annotation-based reconstruction part of the
workflow), put here all the output files of the annotation tool. For
instance, with Pathway Tools, copy-paste the whole PGDB directory
(see below “Annotation-based reconstruction” for more details).</p>
<p><strong>*database*</strong>: if you want to use your own database put in this
folder your database in padmet format, if you have a sbml convert
this file to padmet (see <strong>“FAQ How to convert files to different
formats”</strong>)</p>
<p><strong>*gapfilling/original_output*</strong>: if you run the metabolic network
reconstruction with gap-filling, will contain all the output files
of gap-filling tools before any post-process from AuReMe.</p>
<p><a class="reference internal" href="media/image2.png"><img alt="image1" src="media/image2.png" style="width: 3.04028in; height: 5.33542in;" /></a><strong>*genomic-data*</strong>: the directory in which to put the
genomic data on your studied organism, that is to say either a
Genbank (.gbk) or a proteome (.faa).</p>
<p><strong>*growth_medium*</strong>: description of the set of metabolites that is
available to initiate the metabolism (growth medium), that is to say
the seed compounds (.txt) (see <strong>“FAQ How to manage growth
medium?”</strong>).</p>
<p><strong>*manual_curation*</strong>: all the file to describe the manual curation
you want to apply on your metabolic network (either adding, deleting
or modifying reactions).</p>
<p><strong>*networks*</strong>: all the metabolic networks used or created during
the reconstruction process</p>
<p><strong>*networks* *external_network*</strong>: put here all existing metabolic
networks (.sbml) you want to use. Enables to merge them with the
ones created thanks to other methods.</p>
<p><strong>*networks* *output_annotation_based_reconstruction*</strong>: will
contain the processed network from the annotation based
reconstruction, after the pre-processing of the data from the
<strong>*annotation_based_reconstruction*</strong> directory (if you filled this
one).</p>
<p><strong>*networks* *output_orthology_based_reconstruction*</strong>: will contain
the processed network from the orthology based reconstruction, after
the pre-processing of the data from the
<strong>*orthology_based_reconstruction*</strong> directory (if you have run this
part of the workflow).</p>
<p><strong>*orthology_based_reconstruction*</strong>: if you want to use model
organisms (to run orthology-based reconstruction part of the
workflow), put here the proteome (.faa or .gbk) and the metabolic
network (.sbml) of your model (see below “Orthology-based
reconstruction” for more details).</p>
<p><strong>*targets_compounds*</strong>: description of the set target compounds
(.txt), that is to say metabolites whose production is supposed to
be achieved by the metabolism of the species under study (components
of the biomass reactions or other metabolites).</p>
</div></blockquote>
</div>
<div class="section" id="provide-input-files">
<h3>Provide input files<a class="headerlink" href="#provide-input-files" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><p>First of all, you have to provide to AuReMe all the input files
needed for the different steps you want to run in the workflow. The
steps can be optional or run several times, at different phases of
the process. However, you have to store the input data for each
steps, observing the architecture described above for the
<strong>*bridge*</strong> directory (see <strong>“Data organization Bridge
structure”</strong>).</p>
<p>Here is the list of input you have to provide to run the pre-set
default workflow. If you want to run only part of it, please see the
corresponding sections and the chapter <strong>“How to create your
‘à-la-carte’ workflow”</strong>.</p>
</div></blockquote>
<ul class="simple">
<li><strong>See “Orthology-based reconstruction Inputs”</strong></li>
<li><strong>See “Annotation-based reconstruction Inputs”</strong></li>
<li><strong>External source for reconstruction</strong></li>
</ul>
<blockquote>
<div>If you already have one or several external metabolic networks for
your studied species and you want to improve them, just copy-paste
them (SBML format) in the <strong>*networksexternal_network*</strong> folder.</div></blockquote>
</div>
<div class="section" id="image2check-input-files-validity">
<h3><a class="reference internal" href="media/image3.png"><img alt="image2" src="media/image3.png" style="width: 1.35354in; height: 2.16535in;" /></a>Check input files validity<a class="headerlink" href="#image2check-input-files-validity" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple">
<li>IMPORTANT: Always check the validity of the inputs before running any
workflow task, and after having put every input files needed for the
steps of the workflow. This will verify the format and consistency of
your data for a better quality result. Moreover, it will generate all
the supplementary files needed by the workflow tools and put them
into the corresponding folders. For more information about input
files validity see <strong>“FAQ What is checked in my input files?”</strong>.</li>
</ol>
<blockquote>
<div>For this purpose, use this command:</div></blockquote>
</div>
</div>
<div class="section" id="orthology-based-reconstruction">
<h2>Orthology-based reconstruction<a class="headerlink" href="#orthology-based-reconstruction" title="Permalink to this headline">¶</a></h2>
<div class="section" id="method-pantograph">
<h3>Method: Pantograph<a class="headerlink" href="#method-pantograph" title="Permalink to this headline">¶</a></h3>
<p>Input files:</p>
<p><a class="reference internal" href="media/image4.png"><img alt="image3" src="media/image4.png" style="width: 1.35347in; height: 2.16528in;" /></a>- Required for the orthology-based reconstruction (method:
Pantograph):</p>
<ul class="simple">
<li>Genbank or Proteome of your studied organism (.gbk or .faa)</li>
<li>Genbank or Proteome of your reference organism (.gbk or .faa)</li>
<li>Metabolic network of your reference organism (.sbml)</li>
<li>(option) a dictionary file if genes ids used in metabolic network are</li>
</ul>
<p>different with gbk/faa (.txt)</p>
<p>Result file:</p>
<p>/test</p>
<p>|–orthology_based_reconstruction</p>
<p>| |– <em>model_a</em></p>
<p>| |– <strong>original_output_pantograph_*model_a*.sbml</strong></p>
<p>|– networks</p>
<p>|– orthology_based_reconstruction</p>
<p>|– pantograph</p>
<p>|– <strong>output_pantograph_</strong><em>model_a</em><strong>.sbml</strong></p>
<div class="section" id="inputs">
<h4>Inputs<a class="headerlink" href="#inputs" title="Permalink to this headline">¶</a></h4>
<ol class="arabic simple">
<li>Put all the available genomic data of the studied organism in the
folder <strong>*genomic_data*</strong>, either a Genbank (.gbk) or a Fasta (.faa)
file. WARNING: give them these exact names (respectively):
GBK_study.gbk and FAA_study.faa.</li>
<li>For each reference organism you want to use, create a folder in the
folder <strong>*orthology_based_reconstruction*</strong>. Give it the name of your
model organism (e.g. <strong>*model_a*</strong>).</li>
</ol>
<blockquote>
<div>On a Linux operating system, here is the command to create a new
folder named <strong>*model_a*</strong>:</div></blockquote>
<ol class="arabic simple" start="3">
<li>In each folder, put:<ul>
<li><dl class="first docutils">
<dt>the Genbank file of your model organism, with the exact name</dt>
<dd>GBK_model.gbk</dd>
</dl>
</li>
</ul>
</li>
</ol>
<blockquote>
<div>OR the proteome of your model organism, with the exact name
FAA_model.faa</div></blockquote>
<ul class="simple">
<li><dl class="first docutils">
<dt>the metabolic network of your model organism, with the exact name</dt>
<dd>metabolic_model.sbml</dd>
</dl>
</li>
</ul>
<ol class="arabic" start="4">
<li><div class="first line-block">
<div class="line">The genome (or proteome) and the metabolic network of your model
organism have to contain the same kind genes (or proteins)
identifiers to be comparable. If not enough genes (or proteins) are
in common between the two files, the process will stop to avoid
poor quality data production.</div>
<div class="line">If you want to pursue on the process, please provide a dictionary
file between the gene (or protein) identifiers of these two files.
Name this dictionary <strong>*dict_genes.txt*</strong>. Here is the dictionary
file format asked (tabulation separated values):</div>
</div>
</li>
</ol>
</div>
<div class="section" id="run">
<h4>Run<a class="headerlink" href="#run" title="Permalink to this headline">¶</a></h4>
<ol class="arabic simple" start="5">
<li>Important: Remember to check the validity of the inputs before
running any workflow task. If you want to run only the
orthology-based reconstruction, use now this command:</li>
<li>To run only the orthology-based reconstruction, use this command:</li>
<li>IMPORTANT: Because the metabolic network from the reference organism
could came from different databases, it’s critical to check the
database of each network and if needed convert the network to your
reference database selected (see <strong>“How to use the AuReMe workspace
(default workflow) Define the reference database”</strong>).</li>
</ol>
<blockquote>
<div>The previous command will check the database of the file
output_pantograph_mode_a.sbml, if the database is different for the
reference, use the next command to create a mapping file to metacyc
database. For more information about sbml mapping see <strong>“FAQ How to
map a sbml to another database?”</strong>.</div></blockquote>
</div>
</div>
</div>
<div class="section" id="annotation-based-reconstruction">
<h2>Annotation-based reconstruction<a class="headerlink" href="#annotation-based-reconstruction" title="Permalink to this headline">¶</a></h2>
<div class="section" id="method-pathway-tools">
<h3>Method: Pathway Tools<a class="headerlink" href="#method-pathway-tools" title="Permalink to this headline">¶</a></h3>
<p><a class="reference internal" href="media/image5.png"><img alt="image4" src="media/image5.png" style="width: 1.35347in; height: 2.03774in;" /></a></p>
<p>Input files:</p>
<ul class="simple">
<li>Required for the annotation-based reconstruction (method: Pathway</li>
</ul>
<p>Tools):</p>
<p>The output of Pathway tools (PGDB folder)</p>
<p>Result file:</p>
<p>/test</p>
<p>|– networks</p>
<p>|– annotation_based_reconstruction</p>
<p>|– pathwaytools</p>
<p>|– <strong>output_pathwaytools_</strong><em>genome_a</em><strong>.padmet</strong></p>
<p>|– <strong>output_pathwaytools_</strong><em>genome_b</em><strong>.padmet</strong></p>
<div class="section" id="inputs-1">
<span id="id3"></span><h4>Inputs<a class="headerlink" href="#inputs-1" title="Permalink to this headline">¶</a></h4>
<ol class="arabic simple">
<li>Put the output of Pathway Tools (the whole PGDB directory) in the
folder <strong>*annotation_based_reconstruction*</strong></li>
<li>If you have run several times Pathway Tools and want to use all of
these annotations, just copy-paste the other PGDB folders in the
<strong>*annotation_based_reconstruction*</strong> directory.</li>
</ol>
</div>
<div class="section" id="run-1">
<span id="id4"></span><h4>Run<a class="headerlink" href="#run-1" title="Permalink to this headline">¶</a></h4>
<ol class="arabic simple" start="3">
<li>Important: Remember to check the validity of the inputs before
running any workflow task. If you want to run only the
annotation-based reconstruction, use now this command:</li>
<li>To run only the annotation-based reconstruction, use this command.</li>
</ol>
</div>
</div>
</div>
<div class="section" id="merge-metabolic-networks">
<h2>Merge metabolic networks<a class="headerlink" href="#merge-metabolic-networks" title="Permalink to this headline">¶</a></h2>
<p><a class="reference internal" href="media/image6.png"><img alt="image5" src="media/image6.png" style="width: 1.35347in; height: 1.99306in;" /></a></p>
<p>Input files:</p>
<ul class="simple">
<li>metabolic networks in the <strong>*networks*</strong> directory</li>
</ul>
<p>Result files:</p>
<p>/test</p>
<p>|– netowrks</p>
<p>|– <strong>draft.padmet</strong></p>
<p>To merge all available networks from the <strong>*networks*</strong> directory into
one metabolic network, merging all data on the studied species, run this
command:</p>
<p>Note that you can also add external metabolic network to create the
draft (see <strong>“Data organization Provide input files External source for
reconstruction”</strong>).</p>
<p>IMPORTANT: Before merging your networks, check if not already done if
all the sbml are using the reference database. Also check the
compartment ids used in each of them, delete and change compartment if
need.</p>
<p>For example: if a sbml is using KEGG database but your reference
database is metacyc, you will have to map this sbml to create a mapping
file which will be used automatically in the merging process.</p>
<p>If a sbml contains a compartment id like ‘C_c’ and another contains ‘c’,
although they correspond to the same compartment ‘cytosol’ because of
different ids, a compound in ‘C_c’ is not the same as a compound in ‘c’,
therefore there will be a loss of connectivity in the network. see
<strong>“FAQ How to map a sbml to another database?” and “FAQ How to manage
compartment?”</strong></p>
</div>
<div class="section" id="gap-filling">
<h2>Gap-filling<a class="headerlink" href="#gap-filling" title="Permalink to this headline">¶</a></h2>
<div class="section" id="method-meneco">
<h3>Method: Meneco<a class="headerlink" href="#method-meneco" title="Permalink to this headline">¶</a></h3>
<p><a class="reference internal" href="media/image7.png"><img alt="image6" src="media/image7.png" style="width: 1.35383in; height: 2.16535in;" /></a>Input files:</p>
<ul class="simple">
<li>Required for the gap-filling (method: Meneco):</li>
<li>A metabolic network reference database (.padmet or .sbml)</li>
</ul>
<p>(metacyc 20.5, 22.0, BIGG and ModelSeed are available by default)</p>
<ul class="simple">
<li>Seed and target metabolites (.txt)</li>
<li>A metabolic network to fill (typically created during the previous</li>
</ul>
<p>steps)</p>
<p>Result files:</p>
<p>/test</p>
<p>|– netowrks</p>
<p>|– <em>network_name</em><strong>.sbml</strong></p>
<p>|– <em>network_name</em><strong>.padmet</strong></p>
<p>|– gapfilling</p>
<p>|– original_output</p>
<p>| |– <strong>meneco_output_</strong> <em>network_name</em><strong>.txt</strong></p>
<p>|– <strong>gapfilling_solution_</strong> <em>network_name</em><strong>.csv</strong></p>
<div class="section" id="input">
<h4>Input<a class="headerlink" href="#input" title="Permalink to this headline">¶</a></h4>
<ol class="arabic simple">
<li>You must have selected a reference database to fill-in the potential
gaps in the metabolic network. If it is not done yet, please see
<strong>“Requirements Define the reference database ”</strong></li>
<li>Put the seeds file (named seeds.txt) in the <strong>*growth_medium*</strong>
folder. The seed compounds are the description of the set of
metabolites that is available to initiate the metabolism (growth
medium).</li>
</ol>
<blockquote>
<div>Here is as example of the seed file format:</div></blockquote>
<ol class="arabic simple" start="3">
<li>Set the growth medium using this command:</li>
</ol>
<blockquote>
<div><p>For more details on the medium settings, see <strong>“FAQ How to manage
growth medium?”</strong>.</p>
<p>WARNING: If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be overwritten.</p>
</div></blockquote>
<ol class="arabic simple" start="4">
<li>Put the target file (named targets.txt) in the
<strong>*targets_compounds*</strong> folder. The targets are metabolites whose
production is supposed to be achieved by the metabolism of the
species under study (components of the biomass reactions or other
metabolites).</li>
</ol>
<blockquote>
<div>Here is as example of the seed file format:</div></blockquote>
<ol class="arabic simple" start="5">
<li>You will have to indicate which metabolic network you want to
gap-fill with the Meneco software. If you want to gap-fill a network
created in the previous steps, there is nothing to do. Otherwise, put
the network you want to gap-fill (PADMET format) in the
<strong>*networks*</strong> directory.</li>
</ol>
</div>
<div class="section" id="run-2">
<span id="id5"></span><h4>Run<a class="headerlink" href="#run-2" title="Permalink to this headline">¶</a></h4>
<ol class="arabic simple" start="6">
<li>(optional step) To generate the gap-filling solution run this
command:</li>
</ol>
<blockquote>
<div><p>Note: Do not forget the quotation marks.</p>
<p>It will calculate the gap-filling solution on the <em>network_name</em>
network (in the <strong>*networks*</strong> directory) and put it into the
<strong>*gapfilling*</strong> directory as gapfilling_solution_network_name.csv</p>
</div></blockquote>
<ol class="arabic simple" start="7">
<li>To generate the gap-filled network (and run step 6), run this
command:</li>
</ol>
<blockquote>
<div><p>Note: Do not forget the quotation marks.</p>
<p>It will calculate the gap-filling solution (if it is not yet done)
on the <em>network_name</em> network (in the <strong>*networks*</strong> directory) and
put it into the <strong>*gapfilling*</strong> directory. Then it will generate
the metabolic network (<em>new_network_name</em>), completed with the
gap-filling solution, in the <strong>*networks*</strong> directory.</p>
</div></blockquote>
<p>Note that you can first generate the solution, modify it, then generate
the gap-filled network.</p>
<blockquote>
<div>WARNING: If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be overwritten.</div></blockquote>
</div>
</div>
</div>
<div class="section" id="manual-curation">
<h2><strong>Manual curation</strong><a class="headerlink" href="#manual-curation" title="Permalink to this headline">¶</a></h2>
<p>This step can be done several times and at any moment of the workflow.</p>
<ol class="arabic simple">
<li>Describe the manual curation(s) you want to apply by filling the
corresponding form(s) as explained below.</li>
</ol>
<p>Important note: It is highly recommanded to create a new form file
(.csv) each time you want to apply other changes, in order to keep
tracks of them.</p>
<div class="section" id="add-a-reaction-from-the-database-or-delete-a-reaction-in-a-network">
<h3>Add a reaction from the database or delete a reaction in a network<a class="headerlink" href="#add-a-reaction-from-the-database-or-delete-a-reaction-in-a-network" title="Permalink to this headline">¶</a></h3>
<ol class="loweralpha simple">
<li>Copy from the folder <strong>manual_curationdatatemplate</strong> the file
<strong>reaction_to_add_delete.csv</strong> and paste it into the
<strong>manual_curationdata</strong> directory (this way on Linux operating
systems):</li>
<li>Fill this file (follow the exemple in the template).</li>
</ol>
</div>
<div class="section" id="create-new-reaction-s-to-add-in-a-network">
<h3>Create new reaction(s) to add in a network<a class="headerlink" href="#create-new-reaction-s-to-add-in-a-network" title="Permalink to this headline">¶</a></h3>
<ol class="loweralpha simple">
<li>Copy from the folder <strong>manual_curationdatatemplate</strong> the file
<strong>*reaction_creator.csv*</strong> and paste it into the
<strong>manual_curationdata</strong> directory (this way on Linux operating
systems):</li>
<li>Fill this file (follow the exemple in the template).</li>
</ol>
</div>
<div class="section" id="apply-changes">
<h3>Apply changes<a class="headerlink" href="#apply-changes" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple">
<li>To apply the changes described in the <em>my_form_file.csv</em> form file,
run this command:</li>
</ol>
<blockquote>
<div><p>WARNING: If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be overwritten.</p>
<p>Note that all the manual curations made are stored in history files
in the <strong>manual_curationhistory</strong> directory. You can use them to do
the same corrections on other networks for example.</p>
</div></blockquote>
</div>
</div>
</div>
<div class="section" id="faq">
<h1>FAQ<a class="headerlink" href="#faq" title="Permalink to this headline">¶</a></h1>
<div class="section" id="can-i-have-a-sample-of-aureme">
<h2>Can I have a sample of AuReMe?<a class="headerlink" href="#can-i-have-a-sample-of-aureme" title="Permalink to this headline">¶</a></h2>
<p>To get an overview of AuReMe, you can get a sample by using this
command:</p>
<p>You will get a folder named ‘aureme_sample’ in your bridge directory.
This folder contains all input and output files as if you had run the
entire metabolic network reconstruction workflow for the example files
about <em>Tisochrysis lutea (brown algae)</em>.</p>
<p>Look at the logs file to understand the different commands used in the
reconstruction process.</p>
<p>Note: if you do not want to pollute your log files when testing things
in your sample run, do not forget to use the <strong>quiet</strong> argument in your
command(s) if you wish NOT to store any log, this way:</p>
</div>
<div class="section" id="how-to-convert-files-to-different-formats">
<h2>How to convert files to different formats?<a class="headerlink" href="#how-to-convert-files-to-different-formats" title="Permalink to this headline">¶</a></h2>
<p>The AuReMe workspace natively provides several functions for formats
conversion, through the PADMet Python package. The available convertors
are:</p>
<ul class="simple">
<li>From sbml to padmet format:</li>
</ul>
<blockquote>
<div>This command will convert all sbml in networks folder of
‘<em>test</em>’ to one padmet. If you want to convert one sbml to
padmet format, simply put this file in networks folder of your run
and make sure there is no other sbml file, then run the command. If
you want to merge many sbml to one padmet, add all of them in
networks folder then run the command. Ensure that there is no other
files in network folder before running the command, in the case of
sbml they could be added and in other case a reading error could
occur.</div></blockquote>
<ul class="simple">
<li>From padmet to sbml format:</li>
</ul>
<blockquote>
<div>This command will convert the padmet file <em>my_network.padmet</em> from
networks folder of ‘<em>test</em>’ to create a sbml file
<em>my_network.sbml</em>. By default the sbml level is set to ‘<em>3</em>’,
you can change the default value in the config.txt file or with the
argment LVL (3 or 2)</div></blockquote>
<ul class="simple">
<li>From txt to sbml format:</li>
</ul>
<blockquote>
<div>This command will convert a txt file containing compounds ids to a
sbml file <em>/path/to/txt_file.sbml</em>. The txt file must contain one
compound id by line and optionally the compartment of the id which
by default is ‘c’. Example of file:</div></blockquote>
<ul class="simple">
<li>From GFF/GBK to FAA format:</li>
</ul>
<p><strong>NOTE</strong>: AuReMe integrate some scripts from padmet-utils tools. for
example, gbk_to_faa command use the script
/programs/padmet-utils/connection/gbk_to_faa.py. Not all functions are
encapsulated in AuReMe, there is a lot of scripts that could be helpful.
For more information, see <a class="reference external" href="https://gitlab.inria.fr/maite/padmet-utils">https://gitlab.inria.fr/maite/padmet-utils</a>.</p>
</div>
<div class="section" id="how-to-manage-growth-medium">
<h2>How to manage growth medium?<a class="headerlink" href="#how-to-manage-growth-medium" title="Permalink to this headline">¶</a></h2>
<p>In AuReMe, a compound is defined as a part of the growth medium (or
‘seeds’ for gap-filling tools) if this compound is in the compartment
‘C-BOUNDARY’.</p>
<p><a class="reference internal" href="media/image8.png"><img alt="image7" src="media/image8.png" style="width: 7.08611in; height: 0.38056in;" /></a>The growth medium is linked to the metabolic network by two
reactions, a non-reversible reaction named
‘TransportSeed-<em>compound-id</em>’ which transport a compound of the
growth medium from the compartment ‘C-BOUNDARY’ to the ‘e’
(extra-cellular) and a reversible reaction named
‘ExchangeSeed-<em>compound-id’</em> which exchange the same compound from ‘e’
to the ‘c’ (cytosol). When creating a sbml file, the compounds in the
‘C-BOUNDARY’ compartment will be set as ‘BOUNDARY-CONDITION=TRUE’ to
allow flux (see
<a class="reference external" href="http://sbml.org/Documents/FAQ#What_is_this_.22boundary_condition.22_business.3F">http://sbml.org/Documents/FAQ#What_is_this_.22boundary_condition.22_business.3F</a>).</p>
<p>Note: Some metabolic networks manage the growth medium with a reversible
reaction which consume nothing and produce a compound in the ‘c’
compartment. We chose not to do the same for clarity and because it made
some dedicated tools for metabolic network crash.</p>
<ul class="simple">
<li><dl class="first docutils">
<dt>Get the list of compounds corresponding to the growth medium of a</dt>
<dd>network in padmet format:</dd>
</dl>
</li>
</ul>
<blockquote>
<div>Return a list of compounds or an empty list</div></blockquote>
<ul class="simple">
<li>Set the growth medium of a network in padmet format:</li>
</ul>
<blockquote>
<div>This command will remove the current growth medium if existing, then
create the new growth medium by adding the required reactions as
described before.</div></blockquote>
<ul class="simple">
<li>Delete the growth medium of a network in padmet format:</li>
</ul>
<blockquote>
<div>This function will remove all reactions consuming/producing a
compound in ‘C-BOUNDARY’ compartment.</div></blockquote>
<p>WARNING: If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be overwritten.</p>
</div>
<div class="section" id="how-to-manage-metabolic-network-compartment">
<h2>How to manage metabolic network compartment?<a class="headerlink" href="#how-to-manage-metabolic-network-compartment" title="Permalink to this headline">¶</a></h2>
<p>In a metabolic network a compound can occur in different compartment.
Given a reaction transporting CA<sup>2+</sup> from ‘e’ (extra-cellular
compartment) to ‘c’ (cytosol compartment), the compartments involved are
‘e’ and ‘c’. It is important to properly manage the compartments defined
in a network to ensure a correct connection of the reactions. In some
case metabolic networks can use different id to define a same
compartment like ‘C_c’, ‘C’, ‘c’ for cytosol, merging those networks
could leak to a loss of network connectivity. A reaction producing
CA<sup>2+</sup> in ‘c’ and a reaction consuming CA<sup>2+</sup> in ‘C_c’ are
actually not connected, hence the interest of the metabolic network
compartment management commands of AuReMe.</p>
<ul class="simple">
<li>Get the complete list of compartment from a network in padmet format:</li>
</ul>
<blockquote>
<div>Return a list of compartment or an empty list</div></blockquote>
<ul class="simple">
<li>Change the id of a compartment from a network in padmet format:</li>
</ul>
<blockquote>
<div>This command will change the id of the compartment ‘<em>old_id</em>’ to
‘<em>new_id</em>’. This command is required if different ids are used
to define a same compartment, example changing ‘C_c’ to ‘c’, or
‘C-c’ to ‘c’…</div></blockquote>
<ul class="simple">
<li>Delete the growth medium of a network in padmet format:</li>
</ul>
<blockquote>
<div>This function will remove all reactions consuming/producing a
compound in ‘<em>compart_id</em>’ compartment.</div></blockquote>
<p>WARNING: If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be overwritten.</p>
</div>
<div class="section" id="how-to-manage-the-log-files">
<h2>How to manage the log files?<a class="headerlink" href="#how-to-manage-the-log-files" title="Permalink to this headline">¶</a></h2>
<p>By default, the system registers all the executed commands as a log in
the <strong>*bridgetestlog.txt*</strong> file. The whole output of these commands
will also be stored in another file: the <strong>*bridgetestfull_log.txt*</strong>
file.</p>
<p>If you DO NOT wish to store such logs, you can use the <strong>quiet</strong>
argument in your command(s). For example:</p>
<p>It is possible to re-run a previous command by copying the corresponding
command line in the <strong>*bridgetestlog.txt*</strong> file, and pasting it in the
Docker container terminal.</p>
<p>To be able to reproduce the whole workflow applied in a previous study,
please see the <strong>“FAQ Ho to reproduce studies ”</strong> section.</p>
</div>
<div class="section" id="how-to-reproduce-studies">
<h2>How to reproduce studies?<a class="headerlink" href="#how-to-reproduce-studies" title="Permalink to this headline">¶</a></h2>
<p>If you want to re-run the complete workflow of a pre-run study, built
with AuReMe:</p>
<ul>
<li><p class="first">first of all please create a new study (as described in the
<strong>“Requirements</strong> <strong>Define the reference database ”</strong> section) by
running the init command:</p>
<p>(You can choose any run name you want, except pre-existing runs.
Please, avoid other special character than ‘_’ and numbers)</p>
<p>It generates a new folder named <em>my_run2</em> in the <em>bridge</em> directory.</p>
</li>
<li><p class="first">Now, copy all the input data from the previous study in this new
folder (please, follow the folder architecture described in the
<strong>“Data organization”</strong> section).</p>
</li>
<li><p class="first">Copy also the <strong>*log.txt*</strong> file in the <strong>*bridgemy_run2*</strong>
directory. In this log file, change every occurrence of the previous
run name by <strong>*my_run2*</strong>.</p>
</li>
<li><p class="first">Execute this log file.</p>
</li>
</ul>
</div>
<div class="section" id="how-to-create-a-new-a-la-carte-workflow">
<h2>How to create a new ‘à-la-carte’ workflow?<a class="headerlink" href="#how-to-create-a-new-a-la-carte-workflow" title="Permalink to this headline">¶</a></h2>
<p>If you want to add a new step in the workflow or add a new method, it is
possible to customize AuReMe. For that it is necessary to update the
Makefile in your run. Here is an example of how to do it.</p>
<ul class="simple">
<li>Add a new method:</li>
</ul>
<p>First, install your tool by following the documentation associated. For
the example we will add a new tool for orthology-based reconstruction
‘new_tool’ which use the same input as Pantograph (a metabolic network
in sbml format, a gbk of the reference species and the gbk of the study
species) and generate the same output (a metabolic network in sbml
format).</p>
<p>Secondly we will update the Makefile by adding these lines:</p>
<p>Basically this command says that for each folder in
orthology_based_reconstruction (variable declared in config.txt), if the
expected output is not already created, run new_tool.</p>
<p>Finally, to select this method in your new workflow, change in the file
config.txt the variable ORTHOLOGY_METHOD=pantograph by
ORTHOLOGY_METHOD=new_tool</p>
<ul class="simple">
<li>Add a new step or function:</li>
</ul>
<p>Just update the Makefile by adding a new step and use it with this
command</p>
</div>
<div class="section" id="how-to-choose-another-reference-database">
<h2>How to choose another reference database?<a class="headerlink" href="#how-to-choose-another-reference-database" title="Permalink to this headline">¶</a></h2>
<p>It is possible to select a reference database among several. You can
display the list of all available databases by using this command:</p>
<p>The reference database is needed to:</p>
<ul class="simple">
<li>be able to match all the identifiers of the entities of metabolic
networks</li>
<li>gap-fill the metabolic network in the gap-filling step</li>
</ul>
<p>To select one, replace the corresponding path in the configuration file:
<strong>*config.txt*</strong>, in the <strong>*DATA_BASE*</strong> variable. Or you can comment
the line if you don’t want/can’t use a database. The <strong>*config.txt*</strong>
file is stored at the root of your <strong>*bridge*</strong> folder (see <strong>“Running a
Docker container 4.”</strong>).</p>
</div>
<div class="section" id="what-is-checked-in-my-input-files">
<h2>What is checked in my input files?<a class="headerlink" href="#what-is-checked-in-my-input-files" title="Permalink to this headline">¶</a></h2>
<p>Before running any command in AuReMe, it is highlight recommended to use
the command ‘check_input’. This command checks the validity of the input
files and can also create required files. Concretely this command:</p>
<ul class="simple">
<li>Checks database: If database was specified in the config.txt file
(see the <strong>“FAQ How to choose another reference database studies ”</strong>
section). If so, checks if a sbml version exist and create it on the
other hand.</li>
<li>Checks studied organism data: Search if there is a genbank (gbk/gff)
‘GBK_study.gbk’ and proteome (faa) ‘FAA_study.faa’ in genomic_data
folder. If there is only a genbank, create the proteome (command
‘gbk_to_faa). If there is only the proteome or any of them, just
continue the checking process. Note that the proteome is only
required for the orthology-based reconstruction, method: Pantograph.</li>
<li><a class="reference internal" href="media/image9.png"><img alt="image8" src="media/image9.png" style="width: 6.65625in; height: 0.82014in;" /></a>Checks orthology-based reconstruction data: for each folder
found in ‘orthology_based_reconstruction’ folder checks in each of
them if there is proteome ‘FAA_model.faa’ and a metabolic network
‘metabolic_model.sbml’, if there is no proteome but a genbank file
‘GBK_study.faa’, create the proteome (command ‘gbk_to_faa). Finally,
the command compares the ids of genes/proteins between the proteome
and the metabolic network.</li>
</ul>
<p>If cutoff… important because… dict file to create a new proteome file …</p>
<ul class="simple">
<li>Checks annotation-based reconstruction data: for each folder found in
annotation_based_reconstruction’ folder checks in each of them if
it’s a PGBD from pathway then create (if not already done) a padmet
file ‘<a href="#id12"><span class="problematic" id="id13">output_pathwaytools_</span></a>’folder_name’.padmet in
networks/output_annotation_based_reconstruction folder.</li>
<li>Checks gap-filling data: In order to gap-fill a metabolic network,
Pantograph required as input, a file ‘seeds.sbml’ describing the
seeds (the compounds available for the network), another describing
the targets (the compounds that the network have to be able to
reach), the metabolic network to fill and the database from where to
draw the reactions all in sbml format. It’s possible to start from
txt files for seeds ‘seeds.txt’ and targets ‘targets.txt’, each file
containing the ids of the compounds, one by line. The command will
then convert them to sbml (command ‘compounds_to_sbml’).</li>
</ul>
<p>Note that by default, AuReMe will integrate the artefacts
‘default_artefacts_metacyc_20.0.txt’ to the seeds to create a file
‘seeds_artefacts.txt’ and ‘seeds_artefacts.sbml’. For more information
about the artefacts see <strong>“FAQ What are ‘artefacts’ ”</strong> section</p>
<p>Example:</p>
<p><a href="#id6"><span class="problematic" id="id7">**</span></a>[output] **</p>
<p>INSERT SCREEN FROM check_input log</p>
</div>
<div class="section" id="what-is-the-makefile">
<h2>What is the Makefile?<a class="headerlink" href="#what-is-the-makefile" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="what-is-the-config-txt-file">
<h2>What is the config.txt file?<a class="headerlink" href="#what-is-the-config-txt-file" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="how-to-regenerate-a-new-database-version">
<h2>How to regenerate a new database version?<a class="headerlink" href="#how-to-regenerate-a-new-database-version" title="Permalink to this headline">¶</a></h2>
<p>Voir les notes de Jeanne sur le problème de Sebastian</p>
</div>
<div class="section" id="how-to-map-a-metabolic-network-on-another-database">
<h2>How to map a metabolic network on another database?<a class="headerlink" href="#how-to-map-a-metabolic-network-on-another-database" title="Permalink to this headline">¶</a></h2>
<p>Metabolic networks can be products of varied databases. If you want to
merge efficiently information about metabolic networks coming from
different databases, you will need to map the metabolic network(s) to a
common database. To do so, a solution is provided be AuReMe.</p>
<p>Note: to use this method, the metabolic network to map needs to be in
the SBML format and stored in the <strong>*networks*</strong> folder.</p>
<ul>
<li><div class="first line-block">
<div class="line">First of all, you need to know the origin database of the data. To
recognize the database used in an SBML file, use the <strong>*which_db*</strong>
command:</div>
<div class="line">Example:</div>
</div>
<p><a href="#id8"><span class="problematic" id="id9">**</span></a>[output] **</p>
</li>
<li><p class="first">When you know the origin database of the data, you have to generate
the mapping dictionary from this database to the new one:</p>
<p>Example:</p>
<p><a href="#id10"><span class="problematic" id="id11">**</span></a>[output] **</p>
<p>In this example, the system has found more than just one mapping for
the <em>R_R00494_c</em> reaction and the <em>S_Starch_p</em> compound. It did not
manage to choose between the propositions: the mapping will not be
added to the output mapping. If you want to force the mapping, you
have to modify the mapping file manually.</p>
</li>
<li><p class="first">Once you have created a mapping dictionary file, it will be
automatically applied across the workflow to translate the data.</p>
</li>
</ul>
</div>
<div class="section" id="how-to-generate-reports-on-results">
<h2>How to generate reports on results?<a class="headerlink" href="#how-to-generate-reports-on-results" title="Permalink to this headline">¶</a></h2>
<p>Create reports on the <em>network_name</em> network (in the <strong>*networks*</strong>
directory). The reports is created in the <strong>*analysisreports*</strong>
directory.</p>
<p>Crée 4 fichiers bridge/test/analysis/report/network_name:</p>
<ul class="simple">
<li>All_genes :</li>
</ul>
<p>Id common name linked reactions (;)</p>
<ul class="simple">
<li>All_metabolites</li>
</ul>
<p>dbRef_id common name Produced (p), Consumed (c), Both (cp)</p>
<ul class="simple">
<li>All_pathways</li>
</ul>
<p>dbRef_id common name Number of reaction found Total number of reaction
Ratio</p>
<ul class="simple">
<li>All_reactions</li>
</ul>
<p>nbRef_id common name formula (with ID) formula (with common name) in
pathway associated genes categories</p>
</div>
<div class="section" id="how-to-generate-wiki">
<h2>How to generate Wiki?<a class="headerlink" href="#how-to-generate-wiki" title="Permalink to this headline">¶</a></h2>
<p>Voir la formation de Méziane</p>
<p><a class="reference internal" href="media/image10.png"><img alt="image9" src="media/image10.png" style="width: 1.35383in; height: 2.16535in;" /></a></p>
<ol class="arabic simple">
<li>Create a wiki<ol class="loweralpha">
<li>Create the wiki pages. The pages will be in
analysis/wiki_pages/network_name</li>
</ol>
</li>
</ol>
<p>Wiki_Docker is an image that allows to automatize the creation of wiki
in containers.</p>
<ul class="simple">
<li>Run the next commands from your machine and not from the AuReMe
container.<ol class="loweralpha" start="2">
<li>Download the wiki docker image.</li>
<li>Run and setup a container with wiki docker. Follow the
instructions to setup correctly the wiki.</li>
<li>Send the pages and the configuration to the wiki</li>
</ol>
</li>
</ul>
</div>
<div class="section" id="how-to-connect-to-pathway-tools">
<h2>How to connect to Pathway-tools?<a class="headerlink" href="#how-to-connect-to-pathway-tools" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Create PGDB from output of AuReMe</li>
</ul>
</div>
<div class="section" id="what-are-artefacts">
<h2>What are “artefacts”?<a class="headerlink" href="#what-are-artefacts" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="how-to-process-flux-balance-analysis">
<h2>How to process Flux Balance Analysis?<a class="headerlink" href="#how-to-process-flux-balance-analysis" title="Permalink to this headline">¶</a></h2>
<p>Notes Mez</p>
<p>To set the objective reaction, please see the following FAQ section.</p>
</div>
<div class="section" id="how-to-set-an-objective-reaction">
<h2>How to set an objective reaction?<a class="headerlink" href="#how-to-set-an-objective-reaction" title="Permalink to this headline">¶</a></h2>
<p>Notes Mez</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="index.html" class="btn btn-neutral" title="Welcome to aureme’s documentation!" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Meziane AITE

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    

  

  <script type="text/javascript" src="_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>