

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>FAQ &mdash; aureme  documentation</title>
  

  
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="How to use the AuReMe workspace (default workflow)" href="aureme.html" /> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> aureme
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="docker.html">How to use Docker?</a></li>
<li class="toctree-l1"><a class="reference internal" href="aureme.html">How to use the AuReMe workspace (default workflow)</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">FAQ</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#can-i-have-a-sample-of-aureme">Can I have a sample of AuReMe?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-convert-files-to-different-formats">How to convert files to different formats?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-manage-growth-medium">How to manage growth medium?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-manage-metabolic-network-compartment">How to manage metabolic network compartment?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-are-artefacts">What are “artefacts”?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-explore-the-topology-of-a-metabolic-network">How to explore the topology of a metabolic network?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-manage-the-log-files">How to manage the log files?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-reproduce-studies">How to reproduce studies?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-create-a-new-a-la-carte-workflow">How to create a new ‘à-la-carte’ workflow?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-choose-another-reference-database">How to choose another reference database?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-is-checked-in-my-input-files">What is checked in my input files?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-is-the-makefile">What is the Makefile?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-is-the-config-txt-file">What is the config.txt file?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-regenerate-a-new-database-version">How to regenerate a new database version?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-map-a-metabolic-network-on-another-database">How to map a metabolic network on another database?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-generate-report-on-results">How to generate report on results?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-generate-wiki">How to generate Wiki?</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#requirements">Requirements</a></li>
<li class="toctree-l3"><a class="reference internal" href="#wiki-creation">Wiki creation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#public-and-private-access">Public and private access</a></li>
<li class="toctree-l3"><a class="reference internal" href="#other-wiki-commands">Other wiki commands</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-connect-to-pathway-tools">How to connect to Pathway-tools?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-set-an-objective-reaction">How to set an objective reaction?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-process-flux-balance-analysis">How to process Flux Balance Analysis?</a></li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">aureme</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>FAQ</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/faq.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="faq">
<h1>FAQ<a class="headerlink" href="#faq" title="Permalink to this headline">¶</a></h1>
<div class="section" id="can-i-have-a-sample-of-aureme">
<h2>Can I have a sample of AuReMe?<a class="headerlink" href="#can-i-have-a-sample-of-aureme" title="Permalink to this headline">¶</a></h2>
<p>To get an overview of AuReMe, you can get a sample by using this
command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">sample</span>
</pre></div>
</div>
<p>You will get a folder named ‘aureme_sample’ in your bridge directory (i.e:
/home/data/run_template/aureme_sample). This folder contains all input and o
utput files as if you had run the entire metabolic network reconstruction
workflow for the example files about <em>Tisochrysis lutea</em> (microalgae).
Look at the logs file to understand the different commands used in the
reconstruction process.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>if you do not want to pollute your log files when testing things
in your sample run, do not forget to use the <strong>quiet (-q)</strong>
argument in your command(s) if you wish <strong>not</strong> to store any log,
this way:</p>
<div class="last highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">aureme_sample</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;cmd args&quot;</span> <span class="o">-</span><span class="n">q</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="how-to-convert-files-to-different-formats">
<span id="formats"></span><h2>How to convert files to different formats?<a class="headerlink" href="#how-to-convert-files-to-different-formats" title="Permalink to this headline">¶</a></h2>
<p>The AuReMe workspace natively provides several functions for formats
conversion, through the
<a class="reference external" href="https://pypi.org/project/padmet/">PADMet Python package</a>. The
available convertors are:</p>
<ul>
<li><p class="first">From sbml to padmet format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;draft&quot;</span>
</pre></div>
</div>
</li>
</ul>
<p>This command will convert all sbml in networks folder of ‘<strong>test</strong>’
to one padmet. If you want to convert one sbml to padmet format,
simply put this file in networks folder of your run and make sure
there is no other sbml file nor padmet file, either in networks
directory, or in one of the sub-directory of networks. Then run
the command.</p>
<p>If you want to merge many sbml to one padmet, add all of them in
<strong>networks &gt; external_network</strong> folder then run the command.
Ensure that there is no other sbml nor padmet file, either in
networks directory, or in one of the sub-directory of networks before
running the command. In the case one sbml would be forgotten, it
could add to the resulted <strong>draft.padmet</strong> or another reading error
could occur.</p>
<ul>
<li><p class="first">From padmet to sbml format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;padmet_to_sbml NETWORK=my_network [LVL=3]&quot;</span>
</pre></div>
</div>
</li>
</ul>
<p>This command will convert the padmet file <strong>my_network.padmet</strong> from
networks folder of ‘<strong>test</strong>’ to create a sbml file <strong>my_network.sbml</strong>.
By default the sbml level is set to ‘<strong>3</strong>’, you can change the default
value in the config.txt file or with the argment LVL (3 or 2)</p>
<ul>
<li><p class="first">From txt to sbml format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;compounds_to_sbml CPD=/path/to/file/root_txt_file&quot;</span>
</pre></div>
</div>
</li>
</ul>
<p>This command will convert a txt file containing compounds ids to a
sbml file <strong>/path/to/txt_file.sbml</strong>. The txt file must contain one
compound id by line and optionally the compartment of the id which
by default is ‘c’. Example of file:</p>
<blockquote>
<div><table border="1" class="docutils">
<colgroup>
<col width="100%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><div class="first last line-block">
<div class="line">ATP</div>
<div class="line">ADP</div>
<div class="line">WATER\tC-BOUNDARY</div>
<div class="line">LIGHT\tC-BOUNDARY</div>
</div>
</td>
</tr>
</tbody>
</table>
</div></blockquote>
<ul class="simple">
<li>From GFF/GBK to FAA format:</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>AuReMe integrate some scripts from
<a class="reference external" href="https://github.com/AuReMe/padmet-utils">padmet-utils tools</a>,
for example, gbk_to_faa command use the script
/programs/padmet-utils/connection/gbk_to_faa.py. Not all
functions are encapsulated in AuReMe, there is a lot of scripts
that could be helpful. For more information, see
<a class="reference external" href="https://github.com/AuReMe/padmet-utils">https://github.com/AuReMe/padmet-utils</a>.</p>
<div class="last highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;gbk_to_faa GBK_FILE=/path/to/gbk_file OUTPUT=/path/to/output_file&quot;</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="how-to-manage-growth-medium">
<span id="growth-medium"></span><h2>How to manage growth medium?<a class="headerlink" href="#how-to-manage-growth-medium" title="Permalink to this headline">¶</a></h2>
<p>In AuReMe, a compound is defined as a part of the growth medium (or
‘seeds’ for gap-filling tools) if this compound is in the compartment
‘C-BOUNDARY’.</p>
<a class="reference internal image-reference" href="_images/sbml.png"><img alt="Examples of compartments in a sbml file." src="_images/sbml.png" style="width: 988px; height: 58px;" /></a>
<p>The growth medium is linked to the metabolic network by two reactions,
a non-reversible reaction named ‘<em>TransportSeed-compound-id</em>’ which
transport a compound of the growth medium from the compartment
‘C-BOUNDARY’ to the ‘e’ (extra-cellular) and a reversible reaction named
‘<em>ExchangeSeed-compound-id</em>’ which exchange the same compound from ‘e’
to the ‘c’ (cytosol). When creating a sbml file, the compounds in the
‘C-BOUNDARY’ compartment will be set as ‘BOUNDARY-CONDITION=TRUE’ to
allow flux (see
<a class="reference external" href="http://sbml.org/Documents/FAQ#What_is_this_.22boundary_condition.22_business.3F">http://sbml.org/Documents/FAQ#What_is_this_.22boundary_condition.22_business.3F</a>).</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Some metabolic networks manage the growth medium with a reversible
reaction which consume nothing and produce a compound in the ‘c’
compartment. We chose not to do the same for clarity and because
this metod made crash some dedicated tools for metabolic network .</p>
</div>
<ul>
<li><p class="first">Get the list of compounds corresponding to the growth medium of a
network in padmet format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;get_medium NETWORK=network_name&quot;</span>
</pre></div>
</div>
</li>
</ul>
<blockquote>
<div>Return a list of compounds or an empty list</div></blockquote>
<ul>
<li><p class="first">Set the growth medium of a network in padmet format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;set_medium NETWORK=network_name [NEW_NETWORK=new_network_name]&quot;</span>
</pre></div>
</div>
</li>
</ul>
<p>This command will remove the current growth medium if existing, then
create the new growth medium by adding the required reactions as
described before.</p>
<ul>
<li><p class="first">Delete the growth medium of a network in padmet format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;del_medium NETWORK=network_name [NEW_NETWORK=new_network_name]&quot;</span>
</pre></div>
</div>
</li>
</ul>
<p>This function will remove all reactions consuming/producing a
compound in ‘C-BOUNDARY’ compartment.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be <strong>overwritten</strong>.</p>
</div>
</div>
<div class="section" id="how-to-manage-metabolic-network-compartment">
<span id="compartment"></span><h2>How to manage metabolic network compartment?<a class="headerlink" href="#how-to-manage-metabolic-network-compartment" title="Permalink to this headline">¶</a></h2>
<p>In a metabolic network a compound can occur in different compartment.
Given a reaction transporting CA<sup>2+</sup> from ‘e’ (extra-cellular
compartment) to ‘c’ (cytosol compartment), the compartments involved are
‘e’ and ‘c’. It is important to properly manage the compartments defined
in a network to ensure a correct connection of the reactions. In some
case metabolic networks can use different id to define a same
compartment like ‘C_c’, ‘C’, ‘c’ for cytosol, merging those networks
could leak to a loss of network connectivity. A reaction producing
CA<sup>2+</sup> in ‘c’ and a reaction consuming CA<sup>2+</sup> in ‘C_c’ are
actually not connected, hence the interest of the metabolic network
compartment management commands of AuReMe.</p>
<ul>
<li><p class="first">Get the complete list of compartment from a network in padmet format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;get_compart NETWORK=network_name&quot;</span>
</pre></div>
</div>
</li>
</ul>
<p>Return a list of compartment or an empty list.</p>
<ul>
<li><p class="first">Change the id of a compartment from a network in padmet format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;change_compart NETWORK=network_name OLD=old_id NEW=new_id [NEW_NETWORK=new_network_name]&quot;</span>
</pre></div>
</div>
</li>
</ul>
<p>This command will change the id of the compartment ‘<strong>old_id</strong>’ to
‘<strong>new_id</strong>’. This command is required if different ids are used
to define a same compartment, example changing ‘C_c’ to ‘c’, or
‘C-c’ to ‘c’ …</p>
<ul>
<li><p class="first">Delete the id compartment from a network in padmet format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;del_compart NETWORK=network_name compart=compart_id [NEW_NETWORK=new_network_name]&quot;</span>
</pre></div>
</div>
</li>
</ul>
<p>This function will remove all reactions consuming/producing a
compound in ‘<strong>compart_id</strong>’ compartment.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be <strong>overwritten</strong>.</p>
</div>
</div>
<div class="section" id="what-are-artefacts">
<span id="artefacts"></span><h2>What are “artefacts”?<a class="headerlink" href="#what-are-artefacts" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="59%" />
<col width="41%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><div class="first last line-block">
<div class="line">Meneco is a tool that fill the gaps topologically</div>
<div class="line">in a network, thanks to a reference database (see</div>
<div class="line">the <a class="reference internal" href="aureme.html#meneco"><span class="std std-ref">Method: Meneco</span></a> section). In fact, Meneco cannot</div>
<div class="line">product any other metabolite of an cycle without</div>
<div class="line">initiate it before.</div>
<div class="line">Thereby, artefacts are metabolites allow Meneco to</div>
<div class="line">initiate cycles in a metabolic network.</div>
<div class="line">For example in the picture aside, the Kreps cycle</div>
<div class="line">needs to be initiated with Meneco. A manner to</div>
<div class="line">initiate the Kreps cycle into Meneco is to put the</div>
<div class="line">“citrate” metabolite as one of the “artefacts”</div>
</div>
</td>
<td><img alt="_images/artefacts.jpg" class="first" src="_images/artefacts.jpg" />
<div class="last line-block">
<div class="line">before gap-filling the network</div>
<div class="line">thanks to Meneco.</div>
</div>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="how-to-explore-the-topology-of-a-metabolic-network">
<h2>How to explore the topology of a metabolic network?<a class="headerlink" href="#how-to-explore-the-topology-of-a-metabolic-network" title="Permalink to this headline">¶</a></h2>
<p>A manner of exploring and analyzing the topology of a metabolic network is
to use the <a class="reference external" href="https://pypi.org/project/MeneTools/">MeneTools</a> (Metabolic
Network Topology Tools). Two MeneTools: Menecheck and Menescope are included
in AuReMe. You can run the one or the other individually.</p>
<table border="1" class="docutils">
<colgroup>
<col width="49%" />
<col width="51%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><dl class="first last docutils">
<dt><strong>Input files</strong></dt>
<dd><img alt="_images/menetools_input.png" class="first last" src="_images/menetools_input.png" />
</dd>
</dl>
</td>
<td><dl class="first last docutils">
<dt><strong>Result files</strong></dt>
<dd><img alt="_images/menetools_output.png" class="first last" src="_images/menetools_output.png" />
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>To obtain additional information about the file format of <strong>artefacts.txt</strong>,
<strong>seeds.txt</strong>, and <strong>targets.txt</strong>, please refer to <a class="reference internal" href="aureme.html#gap-filling-input"><span class="std std-ref">Gap-filling input</span></a>
and <a class="reference internal" href="#artefacts"><span class="std std-ref">What are “artefacts”?</span></a> sections.</p>
<ul>
<li><p class="first">Menecheck gives the producibility status using graph-based criteria.
To run Menecheck, use this command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;menecheck NETWORK=network_name&quot;</span>
</pre></div>
</div>
</li>
<li><p class="first">Menescope provides the topologically reachable compounds from seeds (and
artefacts) in a metabolic network. To run Menescope, use this command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;menescope NETWORK=network_name&quot;</span>
</pre></div>
</div>
</li>
</ul>
</div>
<div class="section" id="how-to-manage-the-log-files">
<span id="log-file"></span><h2>How to manage the log files?<a class="headerlink" href="#how-to-manage-the-log-files" title="Permalink to this headline">¶</a></h2>
<p>By default, the system registers all the executed commands as a log in
the <strong>bridge &gt; test &gt; log.txt</strong> file. The whole output of these commands
will also be stored in another file: the <strong>bridge &gt; test &gt; full_log.txt</strong>
file.</p>
<p>If you DO NOT wish to store such logs, you can use the <strong>quiet (-q)</strong>
argument in your command(s). For example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;some_commands&quot;</span> <span class="o">-</span><span class="n">q</span>
</pre></div>
</div>
<p>It is possible to re-run a previous command by copying the corresponding
command line in the <strong>bridge &gt; test &gt; log.txt</strong> file, and pasting it in the
Docker container terminal.</p>
<p>To be able to reproduce the whole workflow applied in a previous study,
please see the <a class="reference internal" href="#reproduce-study"><span class="std std-ref">How to reproduce studies?</span></a> section.</p>
</div>
<div class="section" id="how-to-reproduce-studies">
<span id="reproduce-study"></span><h2>How to reproduce studies?<a class="headerlink" href="#how-to-reproduce-studies" title="Permalink to this headline">¶</a></h2>
<p>If you want to re-run the complete workflow of a pre-run study, built
with AuReMe:</p>
<ul>
<li><p class="first">First of all please create a new study (as described in the
<a class="reference internal" href="aureme.html#new-study"><span class="std std-ref">Start a new study</span></a> section) by running the init command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">init</span><span class="o">=</span><span class="n">my_run2</span>
</pre></div>
</div>
</li>
</ul>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">You can choose any run name you want, except pre-existing runs.
Please, avoid other special characters than ‘_’ and numbers).</p>
</div>
<p>It generates a new folder named <strong>my_run2</strong> in the <strong>bridge</strong> directory.</p>
<ul>
<li><p class="first">Update your <strong>config.txt</strong> file, if it is needed.</p>
</li>
<li><p class="first">Now, copy all the input data from the previous study in this new
folder (please, follow the folder architecture described in the
<a class="reference internal" href="aureme.html#organization"><span class="std std-ref">Data organization</span></a> section).</p>
</li>
<li><p class="first">Copy also the <strong>log.txt</strong> file in the <strong>bridge &gt; my_run2</strong>
directory, rename it (for example as run2.txt), and
<strong>change every occurrence of the previous run name by my_run2</strong>.</p>
</li>
<li><p class="first">Execute the previously created file.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="o">./</span><span class="n">shared</span><span class="o">/</span><span class="n">my_run2</span><span class="o">/</span><span class="n">run2</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
</li>
</ul>
</div>
<div class="section" id="how-to-create-a-new-a-la-carte-workflow">
<span id="a-la-carte"></span><h2>How to create a new ‘à-la-carte’ workflow?<a class="headerlink" href="#how-to-create-a-new-a-la-carte-workflow" title="Permalink to this headline">¶</a></h2>
<p>If you want to add a new step in the workflow or add a new method, it is
possible to customize AuReMe. For that it is necessary to update the
Makefile in your run. Here is an example of how to do it.</p>
<ul class="simple">
<li>Add a new method:</li>
</ul>
<p>First, install your tool by following the documentation associated. For
the example we will add a new tool for orthology-based reconstruction
‘new_tool’ which use the same input as Pantograph (a metabolic network
in sbml format, a gbk of the reference species and the gbk of the study
species) and generate the same output (a metabolic network in sbml
format).</p>
<p>Secondly we will update the Makefile by adding these lines:</p>
<p>Basically this command says that for each folder in
orthology_based_reconstruction (variable declared in config.txt), if the
expected output is not already created, run new_tool.</p>
<p>Finally, to select this method in your new workflow, change in the file
config.txt the variable ORTHOLOGY_METHOD=pantograph by
ORTHOLOGY_METHOD=new_tool</p>
<ul class="simple">
<li>Add a new step or function:</li>
</ul>
<p>Just update the Makefile by adding a new step and use it with this
command</p>
</div>
<div class="section" id="how-to-choose-another-reference-database">
<span id="choose-database"></span><h2>How to choose another reference database?<a class="headerlink" href="#how-to-choose-another-reference-database" title="Permalink to this headline">¶</a></h2>
<p>It is possible to select a reference database among several. You can
display the list of all available databases by using this command:</p>
<p>The reference database is needed to:</p>
<ul class="simple">
<li>be able to match all the identifiers of the entities of metabolic
networks</li>
<li>gap-fill the metabolic network in the gap-filling step</li>
</ul>
<p>To select one, replace the corresponding path in the configuration file:
<strong>*config.txt*</strong>, in the <strong>*DATA_BASE*</strong> variable. Or you can comment
the line if you don’t want/can’t use a database. The <strong>*config.txt*</strong>
file is stored at the root of your <strong>*bridge*</strong> folder (see <a class="reference internal" href="docker.html#run-docker"><span class="std std-ref">Running a Docker container</span></a>).</p>
</div>
<div class="section" id="what-is-checked-in-my-input-files">
<span id="check-inputs"></span><h2>What is checked in my input files?<a class="headerlink" href="#what-is-checked-in-my-input-files" title="Permalink to this headline">¶</a></h2>
<p>Before running any command in AuReMe, it is highlight recommended to use
the command ‘check_input’. This command checks the validity of the input
files and can also create required files. Concretely this command:</p>
<ul class="simple">
<li>Checks database: If database was specified in the config.txt file
(see the <a class="reference internal" href="#choose-database"><span class="std std-ref">How to choose another reference database?</span></a> section). If so, checks if a sbml
version exist and create it on the other hand.</li>
<li>Checks studied organism data: Search if there is a genbank (gbk/gff)
‘GBK_study.gbk’ and proteome (faa) ‘FAA_study.faa’ in genomic_data
folder. If there is only a genbank, create the proteome (command
‘gbk_to_faa). If there is only the proteome or any of them, just
continue the checking process. Note that the proteome is only
required for the orthology-based reconstruction, method: Pantograph.</li>
<li><a href="#id7"><span class="problematic" id="id8">|image8|</span></a>Checks orthology-based reconstruction data: for each folder
found in ‘orthology_based_reconstruction’ folder checks in each of
them if there is proteome ‘FAA_model.faa’ and a metabolic network
‘metabolic_model.sbml’, if there is no proteome but a genbank file
‘GBK_study.faa’, create the proteome (command ‘gbk_to_faa). Finally,
the command compares the ids of genes/proteins between the proteome
and the metabolic network.</li>
</ul>
<p>If cutoff… important because… dict file to create a new proteome file …</p>
<ul class="simple">
<li>Checks annotation-based reconstruction data: for each folder found in
annotation_based_reconstruction’ folder checks in each of them if
it’s a PGBD from pathway then create (if not already done) a padmet
file ‘<a href="#id9"><span class="problematic" id="id10">output_pathwaytools_</span></a>’folder_name’.padmet in
networks/output_annotation_based_reconstruction folder.</li>
<li>Checks gap-filling data: In order to gap-fill a metabolic network,
Pantograph required as input, a file ‘seeds.sbml’ describing the
seeds (the compounds available for the network), another describing
the targets (the compounds that the network have to be able to
reach), the metabolic network to fill and the database from where to
draw the reactions all in sbml format. It’s possible to start from
txt files for seeds ‘seeds.txt’ and targets ‘targets.txt’, each file
containing the ids of the compounds, one by line. The command will
then convert them to sbml (command ‘compounds_to_sbml’).</li>
</ul>
<p>Note that by default, AuReMe will integrate the artefacts
‘default_artefacts_metacyc_20.0.txt’ to the seeds to create a file
‘seeds_artefacts.txt’ and ‘seeds_artefacts.sbml’. For more information
about the artefacts see <a class="reference internal" href="#artefacts"><span class="std std-ref">What are “artefacts”?</span></a> section.</p>
<p>Example:</p>
<p><a href="#id1"><span class="problematic" id="id2">**</span></a>[output] **</p>
<p>INSERT SCREEN FROM check_input log</p>
</div>
<div class="section" id="what-is-the-makefile">
<h2>What is the Makefile?<a class="headerlink" href="#what-is-the-makefile" title="Permalink to this headline">¶</a></h2>
<p>Makefile contient les cmd de AuReMe.
exemple de cmd simple</p>
</div>
<div class="section" id="what-is-the-config-txt-file">
<h2>What is the config.txt file?<a class="headerlink" href="#what-is-the-config-txt-file" title="Permalink to this headline">¶</a></h2>
<p>The <strong>config.txt</strong> is found in the <strong>bridge &gt; test</strong> directory. It contains
all the AuReMe parameters: the name of the selected database, the name of the
various choosen methods, and the default parameters of all programs that
AuReMe needed.</p>
<p>If you want to use either another database or another tool already included in
the AuReMe workspace, modify carefully the <strong>config.txt</strong> file.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">The parameters of the <strong>config.txt</strong> must not be changed unless
you are sure of what you want do!</p>
</div>
</div>
<div class="section" id="how-to-regenerate-a-new-database-version">
<h2>How to regenerate a new database version?<a class="headerlink" href="#how-to-regenerate-a-new-database-version" title="Permalink to this headline">¶</a></h2>
<p>Voir les notes de Jeanne sur le problème de Sebastian</p>
<p>padmet/utils/connexion</p>
</div>
<div class="section" id="how-to-map-a-metabolic-network-on-another-database">
<span id="map-database"></span><h2>How to map a metabolic network on another database?<a class="headerlink" href="#how-to-map-a-metabolic-network-on-another-database" title="Permalink to this headline">¶</a></h2>
<p>Metabolic networks can be products of varied databases. If you want to
merge efficiently information about metabolic networks coming from
different databases, you will need to map the metabolic network(s) to a
common database. To do so, a solution is provided be AuReMe.</p>
<p>Note: to use this method, the metabolic network to map needs to be in
the SBML format and stored in the <strong>*networks*</strong> folder.</p>
<ul>
<li><div class="first line-block">
<div class="line">First of all, you need to know the origin database of the data. To
recognize the database used in an SBML file, use the <strong>*which_db*</strong>
command:</div>
<div class="line">Example:</div>
</div>
<p><a href="#id3"><span class="problematic" id="id4">**</span></a>[output] **</p>
</li>
<li><p class="first">When you know the origin database of the data, you have to generate
the mapping dictionary from this database to the new one:</p>
<p>Example:</p>
<p><a href="#id5"><span class="problematic" id="id6">**</span></a>[output] **</p>
<p>In this example, the system has found more than just one mapping for
the <em>R_R00494_c</em> reaction and the <em>S_Starch_p</em> compound. It did not
manage to choose between the propositions: the mapping will not be
added to the output mapping. If you want to force the mapping, you
have to modify the mapping file manually.</p>
</li>
<li><p class="first">Once you have created a mapping dictionary file, it will be
automatically applied across the workflow to translate the data.</p>
</li>
</ul>
</div>
<div class="section" id="how-to-generate-report-on-results">
<h2>How to generate report on results?<a class="headerlink" href="#how-to-generate-report-on-results" title="Permalink to this headline">¶</a></h2>
<p>Create reports on the <strong>network_name.padmet</strong> file network (in the
<strong>networks</strong> directory).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;report NETWORK=network_name&quot;</span>
</pre></div>
</div>
<p>Four files are created in the <strong>analysis &gt; reports &gt; network_name</strong> directory
thanks to the report command.</p>
<ul>
<li><p class="first">all_genes.csv (has the following format):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="nb">id</span>         <span class="n">Common</span> <span class="n">name</span>   <span class="n">linked</span> <span class="n">reactions</span>
<span class="n">TL_15991</span>   <span class="n">Unknown</span>       <span class="mf">2.3</span><span class="o">.</span><span class="mf">1.180</span><span class="o">-</span><span class="n">RXN</span><span class="p">;</span><span class="n">RXN</span><span class="o">-</span><span class="mi">9535</span>
<span class="n">TL_5857</span>    <span class="n">Unknown</span>       <span class="n">RXN</span><span class="o">-</span><span class="mi">14271</span><span class="p">;</span><span class="n">RXN</span><span class="o">-</span><span class="mi">2425</span>
<span class="n">TL_6475</span>    <span class="n">Unknown</span>       <span class="n">RXN</span><span class="o">-</span><span class="mi">14229</span>
</pre></div>
</div>
<p>If a gene is linked with several reactions, reactions are separated from
<strong>“;”</strong>.</p>
</li>
<li><p class="first">all_metabolites.csv (has the following format):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">dbRef_id</span>       <span class="n">Common</span> <span class="n">name</span>         <span class="n">Produced</span> <span class="p">(</span><span class="n">p</span><span class="p">),</span> <span class="n">Consumed</span> <span class="p">(</span><span class="n">c</span><span class="p">),</span> <span class="n">Both</span> <span class="p">(</span><span class="n">cp</span><span class="p">)</span>
<span class="n">NAD</span><span class="o">-</span><span class="n">P</span><span class="o">-</span><span class="n">OR</span><span class="o">-</span><span class="n">NOP</span>   <span class="n">NAD</span><span class="p">(</span><span class="n">P</span><span class="p">)</span><span class="o">+</span>             <span class="n">cp</span>
<span class="n">THIOCYSTEINE</span>   <span class="n">thiocysteine</span>        <span class="n">p</span>
<span class="n">CPD</span><span class="o">-</span><span class="mi">18346</span>      <span class="n">cis</span><span class="o">-</span><span class="n">vaccenoyl</span><span class="o">-</span><span class="n">CoA</span>   <span class="n">c</span>
</pre></div>
</div>
</li>
<li><p class="first">all_pathways.csv (has the following format):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">dbRef_id</span>     <span class="n">Common</span> <span class="n">name</span>                              <span class="n">Number</span> <span class="n">of</span> <span class="n">reaction</span> <span class="n">found</span>   <span class="n">Total</span> <span class="n">number</span> <span class="n">of</span> <span class="n">reaction</span>   <span class="n">Ratio</span> <span class="p">(</span><span class="n">Reaction</span> <span class="n">found</span> <span class="o">/</span> <span class="n">Total</span><span class="p">)</span>
<span class="n">COA</span><span class="o">-</span><span class="n">PWY</span><span class="o">-</span><span class="mi">1</span>    <span class="n">coenzyme</span> <span class="n">A</span> <span class="n">biosynthesis</span> <span class="n">II</span> <span class="p">(</span><span class="n">mammalian</span><span class="p">)</span>   <span class="mi">1</span>                          <span class="mi">1</span>                          <span class="mf">1.00</span>
<span class="n">PWY</span><span class="o">-</span><span class="mi">4984</span>     <span class="n">urea</span> <span class="n">cycle</span>                               <span class="mi">1</span>                          <span class="mi">5</span>                          <span class="mf">0.20</span>
<span class="n">PWY</span><span class="o">-</span><span class="mi">7821</span>     <span class="n">tunicamycin</span> <span class="n">biosynthesis</span>                 <span class="mi">1</span>                          <span class="mi">9</span>                          <span class="mf">0.11</span>
</pre></div>
</div>
</li>
<li><p class="first">all_reactions.csv (has the following format):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">dbRef_id</span>    <span class="n">Common</span> <span class="n">name</span>                        <span class="n">formula</span> <span class="p">(</span><span class="k">with</span> <span class="nb">id</span><span class="p">)</span>                                                 <span class="n">formula</span> <span class="p">(</span><span class="k">with</span> <span class="n">common</span> <span class="n">name</span><span class="p">)</span>                                                      <span class="ow">in</span> <span class="n">pathways</span>                     <span class="n">associated</span> <span class="n">genes</span>            <span class="n">categories</span>
<span class="n">NDPK</span>        <span class="n">nucleoside</span><span class="o">-</span><span class="n">diphosphate</span> <span class="n">kinase</span>      <span class="mf">1.0</span> <span class="n">ATP</span> <span class="o">+</span> <span class="mf">1.0</span> <span class="n">DADP</span> <span class="o">=&gt;</span> <span class="mf">1.0</span> <span class="n">ADP</span> <span class="o">+</span> <span class="mf">1.0</span> <span class="n">DATP</span>                          <span class="mf">1.0</span> <span class="n">ATP</span> <span class="o">+</span> <span class="mf">1.0</span> <span class="n">dADP</span> <span class="o">=&gt;</span> <span class="mf">1.0</span> <span class="n">ADP</span> <span class="o">+</span> <span class="mf">1.0</span> <span class="n">dATP</span>                                                                        <span class="n">TL_16529</span><span class="p">;</span><span class="n">TL_13128</span>           <span class="n">ORTHOLOGY</span>
<span class="n">RXN</span><span class="o">-</span><span class="mi">15122</span>   <span class="n">ORF</span>                                <span class="mi">1</span> <span class="n">THR</span> <span class="o">=&gt;</span> <span class="mi">1</span> <span class="n">PROTON</span> <span class="o">+</span> <span class="mi">1</span> <span class="n">CPD</span><span class="o">-</span><span class="mi">15056</span> <span class="o">+</span> <span class="mi">1</span> <span class="n">WATER</span>                         <span class="mi">1</span> <span class="n">L</span><span class="o">-</span><span class="n">threonine</span> <span class="o">=&gt;</span> <span class="mi">1</span> <span class="n">H</span><span class="o">+</span> <span class="o">+</span> <span class="mi">1</span> <span class="p">(</span><span class="mi">2</span><span class="n">Z</span><span class="p">)</span><span class="o">-</span><span class="mi">2</span><span class="o">-</span><span class="n">aminobut</span><span class="o">-</span><span class="mi">2</span><span class="o">-</span><span class="n">enoate</span> <span class="o">+</span> <span class="mi">1</span> <span class="n">H2O</span>                      <span class="n">PWY</span><span class="o">-</span><span class="mi">5437</span><span class="p">;</span><span class="n">ILEUSYN</span><span class="o">-</span><span class="n">PWY</span><span class="p">;</span><span class="n">PWY</span><span class="o">-</span><span class="mi">5826</span>   <span class="n">TL_17207</span><span class="p">;</span><span class="n">TL_12535</span><span class="p">;</span><span class="n">TL_8525</span>   <span class="n">ANNOTATION</span><span class="p">;</span><span class="n">ANNOTATION</span><span class="p">;</span><span class="n">ORTHOLOGY</span>
<span class="n">SGPL11</span>      <span class="n">sphinganine</span> <span class="mi">1</span><span class="o">-</span><span class="n">phosphate</span> <span class="n">aldolase</span>   <span class="mf">1.0</span> <span class="n">CPD</span><span class="o">-</span><span class="mi">649</span> <span class="o">=&gt;</span> <span class="mf">1.0</span> <span class="n">PALMITALDEHYDE</span> <span class="o">+</span> <span class="mf">1.0</span> <span class="n">PHOSPHORYL</span><span class="o">-</span><span class="n">ETHANOLAMINE</span>   <span class="mf">1.0</span> <span class="n">sphinganine</span> <span class="mi">1</span><span class="o">-</span><span class="n">phosphate</span> <span class="o">=&gt;</span> <span class="mf">1.0</span> <span class="n">palmitaldehyde</span> <span class="o">+</span> <span class="mf">1.0</span> <span class="n">O</span><span class="o">-</span><span class="n">phosphoethanolamine</span>                                   <span class="n">TL_105</span>                      <span class="n">ORTHOLOGY</span>
</pre></div>
</div>
</li>
</ul>
<blockquote>
<div>In this file, if there are several data in the same field, data are
separated from <strong>“;”</strong>.</div></blockquote>
</div>
<div class="section" id="how-to-generate-wiki">
<h2>How to generate Wiki?<a class="headerlink" href="#how-to-generate-wiki" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="59%" />
<col width="41%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><div class="first line-block">
<div class="line"><strong>Input files</strong></div>
</div>
<img alt="Input files needed to generate a wiki." src="_images/wiki_input.png" />
<div class="line-block">
<div class="line"><strong>Result files</strong></div>
</div>
<blockquote class="last">
<div><img alt="Output directories are generated in AuReMe." src="_images/wiki_output.png" />
</div></blockquote>
</td>
<td><a class="first last reference internal image-reference" href="_images/wiki.png"><img alt="Wiki visualization in Aureme." src="_images/wiki.png" style="width: 202.8px; height: 324.40000000000003px;" /></a>
</td>
</tr>
</tbody>
</table>
<div class="section" id="requirements">
<h3>Requirements<a class="headerlink" href="#requirements" title="Permalink to this headline">¶</a></h3>
<ol class="arabic">
<li><p class="first">Utilize AuReMe, to create the wiki pages from a metabolic network.</p>
<p>An input file <strong>network_name.padmet</strong> inside the <strong>brigde &gt; test &gt; networks</strong>
directory is needed. The wiki pages will be deployed in <strong>brigde &gt; test</strong>
<strong>&gt; analysis &gt; wiki_pages &gt; network_name</strong>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;wiki_pages NETWORK=network_name&quot;</span>
</pre></div>
</div>
</li>
</ol>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Run all the next commands from your machine and not from the AuReMe
container.</p>
</div>
<p>You can use wikis to analyze or visualize your metabolic networks, thanks to
the <a class="reference external" href="https://www.mediawiki.org/wiki/MediaWiki">MediaWiki</a> technology.</p>
<ol class="arabic" start="2">
<li><p class="first">Clone the wiki software within your computer:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shell</span><span class="o">&gt;</span> <span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">AuReMe</span><span class="o">/</span><span class="n">wiki</span><span class="o">-</span><span class="n">metabolic</span><span class="o">-</span><span class="n">network</span><span class="o">.</span><span class="n">git</span>
<span class="n">shell</span><span class="o">&gt;</span> <span class="n">cd</span> <span class="n">wiki</span><span class="o">-</span><span class="n">metabolic</span><span class="o">-</span><span class="n">network</span><span class="o">/</span><span class="n">wiki</span><span class="o">-</span><span class="n">metabolic</span><span class="o">-</span><span class="n">network</span><span class="o">/</span>
<span class="n">shell</span><span class="o">&gt;</span> <span class="n">make</span> <span class="n">init</span>
</pre></div>
</div>
<p>The <strong>wiki-metabolic-network</strong> is now installed on your computer. You can
manage it in using the <a class="reference external" href="https://www.docker.com/">docker.com</a> commands
(see <a class="reference internal" href="docker.html#tips-docker"><span class="std std-ref">Some tips about Docker</span></a>). wiki-metabolic-network is an image that allows to
automatize the creation of wikis in a container.</p>
</li>
</ol>
<ul>
<li><p class="first">Get the name of the wiki container, it will be usefull to run the next
command.</p>
<blockquote>
<div><a class="reference internal image-reference" href="_images/docker_cmd.jpg"><img alt="List of the containers with these usefull for the wiki." src="_images/docker_cmd.jpg" style="width: 1350.0px; height: 126.0px;" /></a>
</div></blockquote>
</li>
</ul>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">For a shake of genericity, in the following steps of this manual,
we will employ the term of <strong>wiki_cont</strong> instead of
<strong>wikimetabolic_mediawiki_1</strong> (the real one you have to write in
your command lines).</p>
</div>
<ul>
<li><p class="first">To enter the wiki container.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shell</span><span class="o">&gt;</span> <span class="n">docker</span> <span class="n">exec</span> <span class="o">-</span><span class="n">it</span> <span class="n">wiki_cont</span> <span class="n">bash</span>
</pre></div>
</div>
</li>
<li><p class="first">To print the commands of the wiki container.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shell</span><span class="o">&gt;</span> <span class="n">docker</span> <span class="n">exec</span> <span class="o">-</span><span class="n">it</span> <span class="n">wiki_cont</span> <span class="n">wiki</span> <span class="o">--</span><span class="n">help</span>
</pre></div>
</div>
</li>
<li><p class="first">Copy the data previously created thanks to AuReMe, in the wiki container.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shell</span><span class="o">&gt;</span> <span class="n">docker</span> <span class="n">cp</span> <span class="o">/</span><span class="n">test</span><span class="o">/</span><span class="n">analysis</span><span class="o">/</span><span class="n">wiki_pages</span><span class="o">/</span><span class="n">network_name</span> <span class="n">wiki_cont</span><span class="p">:</span><span class="o">/</span><span class="n">home</span><span class="o">/</span>
</pre></div>
</div>
</li>
</ul>
</div>
<div class="section" id="wiki-creation">
<span id="wiki-create"></span><h3>Wiki creation<a class="headerlink" href="#wiki-creation" title="Permalink to this headline">¶</a></h3>
<p><strong>Follow the instructions on your terminal.</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shell</span><span class="o">&gt;</span> <span class="n">docker</span> <span class="n">exec</span> <span class="o">-</span><span class="n">ti</span> <span class="n">wiki_cont</span> <span class="n">wiki</span> <span class="o">--</span><span class="n">init</span><span class="o">=</span><span class="n">id_wiki</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/configure_wiki_1.png"><img alt="Instructions you have to follow in order to configure a wiki." src="_images/configure_wiki_1.png" style="width: 702.8px; height: 562.8px;" /></a>
<ol class="arabic">
<li><p class="first">Open your browser at the following address:
<strong>http://localhost/id_wiki/mw-config/index.php</strong>, and press <strong>“Continue”</strong>.</p>
<a class="reference internal image-reference" href="_images/configure_wiki_2.png"><img alt="First step of the instructions of wiki configuration." src="_images/configure_wiki_2.png" style="width: 744.0px; height: 441.59999999999997px;" /></a>
</li>
<li><p class="first">Get the <strong>“Upgrade key”</strong>. The <strong>Upgrade key</strong> is found on the your
terminal. This is a small part extracted from the terminal to locate it
better.</p>
<a class="reference internal image-reference" href="_images/configure_wiki_3.png"><img alt="Second step of the instructions of wiki configuration." src="_images/configure_wiki_3.png" style="width: 339.0px; height: 56.0px;" /></a>
</li>
<li><p class="first">Enter the <strong>“Upgrade key”</strong>, and press <strong>“Continue”</strong>.</p>
<a class="reference internal image-reference" href="_images/configure_wiki_4.png"><img alt="Third step of the instructions of wiki configuration." src="_images/configure_wiki_4.png" style="width: 744.0px; height: 441.59999999999997px;" /></a>
</li>
<li><p class="first">In the page  <strong>“Welcome to MediaWiki”</strong> configuration, just press
<strong>“Continue”</strong>.</p>
<a class="reference internal image-reference" href="_images/configure_wiki_5.png"><img alt="Fourth step of the instructions of wiki configuration." src="_images/configure_wiki_5.png" style="width: 744.0px; height: 616.8px;" /></a>
</li>
<li><p class="first">In the page  <strong>“Database settings”</strong> configuration, just press
<strong>“Continue”</strong>.</p>
<a class="reference internal image-reference" href="_images/configure_wiki_6.png"><img alt="Fifth step of the instructions of wiki configuration." src="_images/configure_wiki_6.png" style="width: 744.0px; height: 474.59999999999997px;" /></a>
</li>
<li><p class="first">In the page  <strong>“Name”</strong> configuration, you have several fields to fill:</p>
<ol class="loweralpha simple">
<li>Name of wiki: <strong>wiki_name</strong></li>
<li>Your username: <strong>admin</strong></li>
<li>Password: Enter a password (it is at least 8 characters).</li>
<li>Password again: Enter the same password.</li>
<li>Email address: jeanne.got[at]irisa.fr (for example)</li>
<li>Please select the phrase: <strong>“I’m bored already, just install the wiki”</strong>.</li>
<li>Press <strong>“Continue”</strong>.</li>
</ol>
<a class="reference internal image-reference" href="_images/configure_wiki_7.png"><img alt="Sixth step of the instructions of wiki configuration." src="_images/configure_wiki_7.png" style="width: 744.0px; height: 702.6px;" /></a>
</li>
<li><p class="first">In the first page of <strong>“Install”</strong>, just press <strong>“Continue</strong>.</p>
<a class="reference internal image-reference" href="_images/configure_wiki_8.png"><img alt="Seventh step of the instructions of wiki configuration." src="_images/configure_wiki_8.png" style="width: 744.0px; height: 367.8px;" /></a>
</li>
<li><p class="first">In the second page of <strong>“Install”</strong>, just press <strong>“Continue</strong>.</p>
<a class="reference internal image-reference" href="_images/configure_wiki_9.png"><img alt="Eighth step of the instructions of wiki configuration." src="_images/configure_wiki_9.png" style="width: 744.0px; height: 367.8px;" /></a>
</li>
<li><p class="first">Do not download the LocalSettings.php file.</p>
</li>
<li><p class="first"><strong>Go back to your terminal, and press “Enter”.</strong> The wiki is now online
and reachable at this link:
<strong>http://localhost/id_wiki/index.php/Main_Page</strong>.</p>
</li>
</ol>
<ul>
<li><p class="first">To send the “wiki pages” (that you previously copied in the wiki_cont
container) on the wiki.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shell</span><span class="o">&gt;</span> <span class="n">docker</span> <span class="n">exec</span> <span class="o">-</span><span class="n">ti</span> <span class="n">wiki_cont</span> <span class="n">wiki_load</span> <span class="o">--</span><span class="n">action</span><span class="o">=</span><span class="n">load</span> <span class="o">--</span><span class="n">url</span><span class="o">=</span><span class="n">http</span><span class="p">:</span><span class="o">//</span><span class="n">localhost</span><span class="o">/</span><span class="n">id_wiki</span><span class="o">/</span><span class="n">api</span><span class="o">.</span><span class="n">php</span> <span class="o">--</span><span class="n">user</span><span class="o">=</span><span class="n">admin</span> <span class="o">--</span><span class="n">password</span><span class="o">=</span><span class="n">my_password</span> <span class="o">--</span><span class="n">wikipage</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">network_name</span> <span class="o">--</span><span class="n">bots</span><span class="o">=</span><span class="mi">2</span> <span class="o">-</span><span class="n">v</span>
</pre></div>
</div>
<p>Here <strong>“bots”</strong> is the number of CPUs are allocated to make this task.</p>
</li>
<li><p class="first">Now wiki pages are accessible on <strong>http://localhost/id_wiki/index.php</strong>.
The following picture shows some functionalities of the wiki.</p>
<a class="reference internal image-reference" href="_images/wiki_func.jpg"><img alt="Some functionalities of the wiki" src="_images/wiki_func.jpg" style="width: 832.0px; height: 718.0px;" /></a>
</li>
</ul>
</div>
<div class="section" id="public-and-private-access">
<h3>Public and private access<a class="headerlink" href="#public-and-private-access" title="Permalink to this headline">¶</a></h3>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">By default, a <strong>“public access”</strong> wiki is created. A wiki with a
public access means, a wiki which everyone is allowed to access and
to edit it on condition that she/he has an account on this wiki.</p>
</div>
<ul>
<li><p class="first">To deploy a wiki with a <strong>“private access”</strong>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shell</span><span class="o">&gt;</span> <span class="n">docker</span> <span class="n">exec</span> <span class="o">-</span><span class="n">ti</span> <span class="n">wiki_cont</span> <span class="n">wiki</span> <span class="o">--</span><span class="n">init</span><span class="o">=</span><span class="n">id_wiki</span> <span class="o">--</span><span class="n">access</span><span class="o">=</span><span class="n">private</span>
</pre></div>
</div>
<p>Then, see the <a class="reference internal" href="#wiki-create"><span class="std std-ref">Wiki creation</span></a> section. A wiki with a <strong>“private access”</strong>
is preventing access and editing for non-user. It also prevent account
creation. It is useful to manage confidential data.</p>
</li>
<li><p class="first">To modify the access of a wiki already created.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shell</span><span class="o">&gt;</span> <span class="n">docker</span> <span class="n">exec</span> <span class="o">-</span><span class="n">ti</span> <span class="n">wiki_cont</span> <span class="n">wiki</span> <span class="o">--</span><span class="nb">id</span><span class="o">=</span><span class="n">id_wiki</span> <span class="o">--</span><span class="n">access</span><span class="o">=</span><span class="n">private</span>
<span class="n">shell</span><span class="o">&gt;</span> <span class="n">docker</span> <span class="n">exec</span> <span class="o">-</span><span class="n">ti</span> <span class="n">wiki_cont</span> <span class="n">wiki</span> <span class="o">--</span><span class="nb">id</span><span class="o">=</span><span class="n">id_wiki</span> <span class="o">--</span><span class="n">access</span><span class="o">=</span><span class="n">public</span>
</pre></div>
</div>
</li>
</ul>
</div>
<div class="section" id="other-wiki-commands">
<h3>Other wiki commands<a class="headerlink" href="#other-wiki-commands" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">To list all deployed wiki use.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shell</span><span class="o">&gt;</span> <span class="n">docker</span> <span class="n">exec</span> <span class="o">-</span><span class="n">ti</span> <span class="n">wiki_cont</span> <span class="n">wiki</span> <span class="o">--</span><span class="nb">all</span>
<span class="n">All</span> <span class="n">deployed</span> <span class="n">wiki</span><span class="p">:</span>
        <span class="n">C_elegans</span>
        <span class="n">E_siliculosus</span>
        <span class="n">id_wiki</span>
        <span class="n">S_cerevisiae</span>
</pre></div>
</div>
</li>
<li><p class="first">To remove a wiki use.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shell</span><span class="o">&gt;</span> <span class="n">docker</span> <span class="n">exec</span> <span class="o">-</span><span class="n">ti</span> <span class="n">wiki_cont</span> <span class="n">wiki</span> <span class="o">--</span><span class="nb">id</span><span class="o">=</span><span class="n">id_wiki</span> <span class="o">--</span><span class="n">remove</span>
<span class="n">Removing</span> <span class="n">wiki</span> <span class="n">id_wiki</span>
<span class="n">Removing</span> <span class="n">wiki</span> <span class="n">folder</span>
<span class="mi">97</span> <span class="n">tables</span> <span class="n">to</span> <span class="n">drop</span>
</pre></div>
</div>
<p>It removes the “id_wiki” from wiki_folders and removes tables from database
which start with prefix id_wiki.</p>
</li>
<li><p class="first">To reset a wiki use.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">shell</span><span class="o">&gt;</span> <span class="n">docker</span> <span class="n">exec</span> <span class="o">-</span><span class="n">ti</span> <span class="n">wiki_cont</span> <span class="n">wiki</span> <span class="o">--</span><span class="nb">id</span><span class="o">=</span><span class="n">id_wiki</span> <span class="o">--</span><span class="n">clean</span>
<span class="n">id_wiki_page</span> <span class="n">table</span> <span class="n">to</span> <span class="n">empty</span>
</pre></div>
</div>
<p>It only remove all the pages of the specified wiki. It keeps tables and
folder associated with this wiki.</p>
</li>
</ul>
</div>
</div>
<div class="section" id="how-to-connect-to-pathway-tools">
<h2>How to connect to Pathway-tools?<a class="headerlink" href="#how-to-connect-to-pathway-tools" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Create PGDB from output of AuReMe</li>
</ul>
</div>
<div class="section" id="how-to-set-an-objective-reaction">
<span id="objective-reaction"></span><h2>How to set an objective reaction?<a class="headerlink" href="#how-to-set-an-objective-reaction" title="Permalink to this headline">¶</a></h2>
<p>To add a biomass reaction to a network, see the <a class="reference internal" href="aureme.html#create-new-reaction"><span class="std std-ref">Create new reaction(s) to add in a network</span></a>
section. Once the biomass is included in the network, you have to set the
biomass as objective function.</p>
<p>Apply this command to the <strong>network_name.padmet</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;set_fba ID=reaction_name NETWORK=network_name&quot;</span>
</pre></div>
</div>
<p>It creates the <strong>network_name.sbml</strong> file with reaction_name as the objective
function. To continue the analyzis of the network_name, see the <a class="reference internal" href="#fba"><span class="std std-ref">How to process Flux Balance Analysis?</span></a>
section.</p>
</div>
<div class="section" id="how-to-process-flux-balance-analysis">
<span id="fba"></span><h2>How to process Flux Balance Analysis?<a class="headerlink" href="#how-to-process-flux-balance-analysis" title="Permalink to this headline">¶</a></h2>
<p>AuReMe evaluate the flux balance analyzis of the biological network, thanks to
the <a class="reference external" href="https://pypi.org/project/cobra/">cobrapy Python package</a>.
Before calculating the flux balance analysis of a network:</p>
<ol class="loweralpha simple">
<li>you may have to add the biomass to a network in reporting to the
<a class="reference internal" href="aureme.html#create-new-reaction"><span class="std std-ref">Create new reaction(s) to add in a network</span></a> section,</li>
<li>you have to set the biomass as an objective reaction, please refer to the
<a class="reference internal" href="#objective-reaction"><span class="std std-ref">How to set an objective reaction?</span></a> section.</li>
</ol>
<p>To compute the flux balance analyzis of the <strong>network_name.sbml</strong> file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;summary NETWORK=network_name&quot;</span>
</pre></div>
</div>
<p>Two files: <strong>network_name.txt</strong> and <strong>network_name_log.txt</strong> are generated
in the <strong>analysis &gt; flux_analysis</strong> directory. The first file
(<strong>network_name.txt</strong>) summarizes te network, then it get the list of
productible and unproductible targets. For each productible target, the flux
balance analysis is given. The growth rate of the network is also provided.
Here is an example of a <strong>network_name.txt</strong> format:</p>
<a class="reference internal image-reference" href="_images/fba_file.png"><img alt="Format of network_name.txt" class="align-center" src="_images/fba_file.png" style="width: 338.40000000000003px; height: 500.40000000000003px;" /></a>
<p>The second file (<strong>network_name_log.txt</strong>) supplies all the warnings produced
computing the flux balance analyzis.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="aureme.html" class="btn btn-neutral" title="How to use the AuReMe workspace (default workflow)" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Meziane AITE

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    
    
      <script type="text/javascript">
          var DOCUMENTATION_OPTIONS = {
              URL_ROOT:'./',
              VERSION:'',
              LANGUAGE:'None',
              COLLAPSE_INDEX:false,
              FILE_SUFFIX:'.html',
              HAS_SOURCE:  true,
              SOURCELINK_SUFFIX: '.txt'
          };
      </script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    

  

  <script type="text/javascript" src="_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>