

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>FAQ &mdash; aureme  documentation</title>
  

  
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="How to use the AuReMe workspace (default workflow)" href="aureme.html" /> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> aureme
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="docker.html">How to use Docker?</a></li>
<li class="toctree-l1"><a class="reference internal" href="aureme.html">How to use the AuReMe workspace (default workflow)</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">FAQ</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#can-i-have-a-sample-of-aureme">Can I have a sample of AuReMe?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-convert-files-to-different-formats">How to convert files to different formats?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-manage-growth-medium">How to manage growth medium?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-manage-metabolic-network-compartment">How to manage metabolic network compartment?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-manage-the-log-files">How to manage the log files?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-reproduce-studies">How to reproduce studies?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-create-a-new-a-la-carte-workflow">How to create a new ‘à-la-carte’ workflow?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-choose-another-reference-database">How to choose another reference database?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-is-checked-in-my-input-files">What is checked in my input files?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-is-the-makefile">What is the Makefile?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-is-the-config-txt-file">What is the config.txt file?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-regenerate-a-new-database-version">How to regenerate a new database version?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-map-a-metabolic-network-on-another-database">How to map a metabolic network on another database?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-generate-reports-on-results">How to generate reports on results?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-generate-wiki">How to generate Wiki?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-connect-to-pathway-tools">How to connect to Pathway-tools?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#what-are-artefacts">What are “artefacts”?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-process-flux-balance-analysis">How to process Flux Balance Analysis?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#how-to-set-an-objective-reaction">How to set an objective reaction?</a></li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">aureme</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>FAQ</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/faq.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="faq">
<h1>FAQ<a class="headerlink" href="#faq" title="Permalink to this headline">¶</a></h1>
<div class="section" id="can-i-have-a-sample-of-aureme">
<h2>Can I have a sample of AuReMe?<a class="headerlink" href="#can-i-have-a-sample-of-aureme" title="Permalink to this headline">¶</a></h2>
<p>To get an overview of AuReMe, you can get a sample by using this
command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">sample</span>
</pre></div>
</div>
<p>You will get a folder named ‘aureme_sample’ in your bridge directory (i.e:
/home/data/run_template/aureme_sample). This folder contains all input and o
utput files as if you had run the entire metabolic network reconstruction
workflow for the example files about <em>Tisochrysis lutea</em> (microalgae).
Look at the logs file to understand the different commands used in the
reconstruction process.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>if you do not want to pollute your log files when testing things
in your sample run, do not forget to use the <strong>quiet (-q)</strong>
argument in your command(s) if you wish <strong>not</strong> to store any log,
this way:</p>
<div class="last highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">aureme_sample</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;cmd args&quot;</span> <span class="o">-</span><span class="n">q</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="how-to-convert-files-to-different-formats">
<span id="formats"></span><h2>How to convert files to different formats?<a class="headerlink" href="#how-to-convert-files-to-different-formats" title="Permalink to this headline">¶</a></h2>
<p>The AuReMe workspace natively provides several functions for formats
conversion, through the PADMet Python package. The available convertors
are:</p>
<ul class="simple">
<li>From sbml to padmet format:</li>
</ul>
<blockquote>
<div>This command will convert all sbml in networks folder of
‘<em>test</em>’ to one padmet. If you want to convert one sbml to
padmet format, simply put this file in networks folder of your run
and make sure there is no other sbml file, then run the command. If
you want to merge many sbml to one padmet, add all of them in
networks folder then run the command. Ensure that there is no other
files in network folder before running the command, in the case of
sbml they could be added and in other case a reading error could
occur.</div></blockquote>
<ul class="simple">
<li>From padmet to sbml format:</li>
</ul>
<blockquote>
<div>This command will convert the padmet file <em>my_network.padmet</em> from
networks folder of ‘<em>test</em>’ to create a sbml file
<em>my_network.sbml</em>. By default the sbml level is set to ‘<em>3</em>’,
you can change the default value in the config.txt file or with the
argment LVL (3 or 2)</div></blockquote>
<ul class="simple">
<li>From txt to sbml format:</li>
</ul>
<blockquote>
<div>This command will convert a txt file containing compounds ids to a
sbml file <em>/path/to/txt_file.sbml</em>. The txt file must contain one
compound id by line and optionally the compartment of the id which
by default is ‘c’. Example of file:</div></blockquote>
<ul class="simple">
<li>From GFF/GBK to FAA format:</li>
</ul>
<p><strong>NOTE</strong>: AuReMe integrate some scripts from padmet-utils tools. for
example, gbk_to_faa command use the script
/programs/padmet-utils/connection/gbk_to_faa.py. Not all functions are
encapsulated in AuReMe, there is a lot of scripts that could be helpful.
For more information, see
<a class="reference external" href="https://github.com/AuReMe/padmet-utils">https://github.com/AuReMe/padmet-utils</a>.</p>
</div>
<div class="section" id="how-to-manage-growth-medium">
<span id="growth-medium"></span><h2>How to manage growth medium?<a class="headerlink" href="#how-to-manage-growth-medium" title="Permalink to this headline">¶</a></h2>
<p>In AuReMe, a compound is defined as a part of the growth medium (or
‘seeds’ for gap-filling tools) if this compound is in the compartment
‘C-BOUNDARY’.</p>
<p><a class="reference internal" href="media/image8.png"><img alt="image7" src="media/image8.png" style="width: 7.08611in; height: 0.38056in;" /></a>The growth medium is linked to the metabolic network by two
reactions, a non-reversible reaction named
‘TransportSeed-<em>compound-id</em>’ which transport a compound of the
growth medium from the compartment ‘C-BOUNDARY’ to the ‘e’
(extra-cellular) and a reversible reaction named
‘ExchangeSeed-<em>compound-id’</em> which exchange the same compound from ‘e’
to the ‘c’ (cytosol). When creating a sbml file, the compounds in the
‘C-BOUNDARY’ compartment will be set as ‘BOUNDARY-CONDITION=TRUE’ to
allow flux (see
<a class="reference external" href="http://sbml.org/Documents/FAQ#What_is_this_.22boundary_condition.22_business.3F">http://sbml.org/Documents/FAQ#What_is_this_.22boundary_condition.22_business.3F</a>).</p>
<p>Note: Some metabolic networks manage the growth medium with a reversible
reaction which consume nothing and produce a compound in the ‘c’
compartment. We chose not to do the same for clarity and because it made
some dedicated tools for metabolic network crash.</p>
<ul class="simple">
<li><dl class="first docutils">
<dt>Get the list of compounds corresponding to the growth medium of a</dt>
<dd>network in padmet format:</dd>
</dl>
</li>
</ul>
<blockquote>
<div>Return a list of compounds or an empty list</div></blockquote>
<ul class="simple">
<li>Set the growth medium of a network in padmet format:</li>
</ul>
<blockquote>
<div>This command will remove the current growth medium if existing, then
create the new growth medium by adding the required reactions as
described before.</div></blockquote>
<ul class="simple">
<li>Delete the growth medium of a network in padmet format:</li>
</ul>
<blockquote>
<div>This function will remove all reactions consuming/producing a
compound in ‘C-BOUNDARY’ compartment.</div></blockquote>
<p>WARNING: If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be overwritten.</p>
</div>
<div class="section" id="how-to-manage-metabolic-network-compartment">
<span id="compartment"></span><h2>How to manage metabolic network compartment?<a class="headerlink" href="#how-to-manage-metabolic-network-compartment" title="Permalink to this headline">¶</a></h2>
<p>In a metabolic network a compound can occur in different compartment.
Given a reaction transporting CA<sup>2+</sup> from ‘e’ (extra-cellular
compartment) to ‘c’ (cytosol compartment), the compartments involved are
‘e’ and ‘c’. It is important to properly manage the compartments defined
in a network to ensure a correct connection of the reactions. In some
case metabolic networks can use different id to define a same
compartment like ‘C_c’, ‘C’, ‘c’ for cytosol, merging those networks
could leak to a loss of network connectivity. A reaction producing
CA<sup>2+</sup> in ‘c’ and a reaction consuming CA<sup>2+</sup> in ‘C_c’ are
actually not connected, hence the interest of the metabolic network
compartment management commands of AuReMe.</p>
<ul class="simple">
<li>Get the complete list of compartment from a network in padmet format:</li>
</ul>
<blockquote>
<div>Return a list of compartment or an empty list</div></blockquote>
<ul class="simple">
<li>Change the id of a compartment from a network in padmet format:</li>
</ul>
<blockquote>
<div>This command will change the id of the compartment ‘<em>old_id</em>’ to
‘<em>new_id</em>’. This command is required if different ids are used
to define a same compartment, example changing ‘C_c’ to ‘c’, or
‘C-c’ to ‘c’…</div></blockquote>
<ul class="simple">
<li>Delete the growth medium of a network in padmet format:</li>
</ul>
<blockquote>
<div>This function will remove all reactions consuming/producing a
compound in ‘<em>compart_id</em>’ compartment.</div></blockquote>
<p>WARNING: If you don’t precise any <strong>NEW_NETWORK</strong> name, the current
network will be overwritten.</p>
</div>
<div class="section" id="how-to-manage-the-log-files">
<span id="log-file"></span><h2>How to manage the log files?<a class="headerlink" href="#how-to-manage-the-log-files" title="Permalink to this headline">¶</a></h2>
<p>By default, the system registers all the executed commands as a log in
the <strong>bridge &gt; test &gt; log.txt</strong> file. The whole output of these commands
will also be stored in another file: the <strong>bridge &gt; test &gt; full_log.txt</strong>
file.</p>
<p>If you DO NOT wish to store such logs, you can use the <strong>quiet (-q)</strong>
argument in your command(s). For example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">run</span><span class="o">=</span><span class="n">test</span> <span class="o">--</span><span class="n">cmd</span><span class="o">=</span><span class="s2">&quot;some_commands&quot;</span> <span class="o">-</span><span class="n">q</span>
</pre></div>
</div>
<p>It is possible to re-run a previous command by copying the corresponding
command line in the <strong>bridge &gt; test &gt; log.txt</strong> file, and pasting it in the
Docker container terminal.</p>
<p>To be able to reproduce the whole workflow applied in a previous study,
please see the <a class="reference internal" href="#reproduce-study"><span class="std std-ref">How to reproduce studies?</span></a> section.</p>
</div>
<div class="section" id="how-to-reproduce-studies">
<span id="reproduce-study"></span><h2>How to reproduce studies?<a class="headerlink" href="#how-to-reproduce-studies" title="Permalink to this headline">¶</a></h2>
<p>If you want to re-run the complete workflow of a pre-run study, built
with AuReMe:</p>
<ul>
<li><p class="first">First of all please create a new study (as described in the
<a class="reference internal" href="aureme.html#new-study"><span class="std std-ref">Start a new study</span></a> section) by running the init command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">aureme</span><span class="o">&gt;</span> <span class="n">aureme</span> <span class="o">--</span><span class="n">init</span><span class="o">=</span><span class="n">my_run2</span>
</pre></div>
</div>
</li>
</ul>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">You can choose any run name you want, except pre-existing runs.
Please, avoid other special characters than ‘_’ and numbers).</p>
</div>
<p>It generates a new folder named <strong>my_run2</strong> in the <strong>bridge</strong> directory.</p>
<ul>
<li><p class="first">Update your <strong>config.txt</strong> file, if it is needed.</p>
</li>
<li><p class="first">Now, copy all the input data from the previous study in this new
folder (please, follow the folder architecture described in the
<a class="reference internal" href="aureme.html#organization"><span class="std std-ref">Data organization</span></a> section).</p>
</li>
<li><p class="first">Copy also the <strong>log.txt</strong> file in the <strong>bridge &gt; my_run2</strong>
directory, rename it (for example as run2.txt), and
<strong>change every occurrence of the previous run name by my_run2</strong>.</p>
</li>
<li><p class="first">Execute the previously created file.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">au</span><span class="o">&gt;</span> <span class="o">./</span><span class="n">shared</span><span class="o">/</span><span class="n">my_run2</span><span class="o">/</span><span class="n">run2</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
</li>
</ul>
</div>
<div class="section" id="how-to-create-a-new-a-la-carte-workflow">
<span id="a-la-carte"></span><h2>How to create a new ‘à-la-carte’ workflow?<a class="headerlink" href="#how-to-create-a-new-a-la-carte-workflow" title="Permalink to this headline">¶</a></h2>
<p>If you want to add a new step in the workflow or add a new method, it is
possible to customize AuReMe. For that it is necessary to update the
Makefile in your run. Here is an example of how to do it.</p>
<ul class="simple">
<li>Add a new method:</li>
</ul>
<p>First, install your tool by following the documentation associated. For
the example we will add a new tool for orthology-based reconstruction
‘new_tool’ which use the same input as Pantograph (a metabolic network
in sbml format, a gbk of the reference species and the gbk of the study
species) and generate the same output (a metabolic network in sbml
format).</p>
<p>Secondly we will update the Makefile by adding these lines:</p>
<p>Basically this command says that for each folder in
orthology_based_reconstruction (variable declared in config.txt), if the
expected output is not already created, run new_tool.</p>
<p>Finally, to select this method in your new workflow, change in the file
config.txt the variable ORTHOLOGY_METHOD=pantograph by
ORTHOLOGY_METHOD=new_tool</p>
<ul class="simple">
<li>Add a new step or function:</li>
</ul>
<p>Just update the Makefile by adding a new step and use it with this
command</p>
</div>
<div class="section" id="how-to-choose-another-reference-database">
<span id="choose-database"></span><h2>How to choose another reference database?<a class="headerlink" href="#how-to-choose-another-reference-database" title="Permalink to this headline">¶</a></h2>
<p>It is possible to select a reference database among several. You can
display the list of all available databases by using this command:</p>
<p>The reference database is needed to:</p>
<ul class="simple">
<li>be able to match all the identifiers of the entities of metabolic
networks</li>
<li>gap-fill the metabolic network in the gap-filling step</li>
</ul>
<p>To select one, replace the corresponding path in the configuration file:
<strong>*config.txt*</strong>, in the <strong>*DATA_BASE*</strong> variable. Or you can comment
the line if you don’t want/can’t use a database. The <strong>*config.txt*</strong>
file is stored at the root of your <strong>*bridge*</strong> folder (see <a class="reference internal" href="docker.html#run-docker"><span class="std std-ref">Running a Docker container</span></a>).</p>
</div>
<div class="section" id="what-is-checked-in-my-input-files">
<span id="check-inputs"></span><h2>What is checked in my input files?<a class="headerlink" href="#what-is-checked-in-my-input-files" title="Permalink to this headline">¶</a></h2>
<p>Before running any command in AuReMe, it is highlight recommended to use
the command ‘check_input’. This command checks the validity of the input
files and can also create required files. Concretely this command:</p>
<ul class="simple">
<li>Checks database: If database was specified in the config.txt file
(see the <a class="reference internal" href="#choose-database"><span class="std std-ref">How to choose another reference database?</span></a> section). If so, checks if a sbml
version exist and create it on the other hand.</li>
<li>Checks studied organism data: Search if there is a genbank (gbk/gff)
‘GBK_study.gbk’ and proteome (faa) ‘FAA_study.faa’ in genomic_data
folder. If there is only a genbank, create the proteome (command
‘gbk_to_faa). If there is only the proteome or any of them, just
continue the checking process. Note that the proteome is only
required for the orthology-based reconstruction, method: Pantograph.</li>
<li><a class="reference internal" href="media/image9.png"><img alt="image8" src="media/image9.png" style="width: 6.65625in; height: 0.82014in;" /></a>Checks orthology-based reconstruction data: for each folder
found in ‘orthology_based_reconstruction’ folder checks in each of
them if there is proteome ‘FAA_model.faa’ and a metabolic network
‘metabolic_model.sbml’, if there is no proteome but a genbank file
‘GBK_study.faa’, create the proteome (command ‘gbk_to_faa). Finally,
the command compares the ids of genes/proteins between the proteome
and the metabolic network.</li>
</ul>
<p>If cutoff… important because… dict file to create a new proteome file …</p>
<ul class="simple">
<li>Checks annotation-based reconstruction data: for each folder found in
annotation_based_reconstruction’ folder checks in each of them if
it’s a PGBD from pathway then create (if not already done) a padmet
file ‘<a href="#id7"><span class="problematic" id="id8">output_pathwaytools_</span></a>’folder_name’.padmet in
networks/output_annotation_based_reconstruction folder.</li>
<li>Checks gap-filling data: In order to gap-fill a metabolic network,
Pantograph required as input, a file ‘seeds.sbml’ describing the
seeds (the compounds available for the network), another describing
the targets (the compounds that the network have to be able to
reach), the metabolic network to fill and the database from where to
draw the reactions all in sbml format. It’s possible to start from
txt files for seeds ‘seeds.txt’ and targets ‘targets.txt’, each file
containing the ids of the compounds, one by line. The command will
then convert them to sbml (command ‘compounds_to_sbml’).</li>
</ul>
<p>Note that by default, AuReMe will integrate the artefacts
‘default_artefacts_metacyc_20.0.txt’ to the seeds to create a file
‘seeds_artefacts.txt’ and ‘seeds_artefacts.sbml’. For more information
about the artefacts see <a class="reference internal" href="#artefacts"><span class="std std-ref">What are “artefacts”?</span></a> section.</p>
<p>Example:</p>
<p><a href="#id1"><span class="problematic" id="id2">**</span></a>[output] **</p>
<p>INSERT SCREEN FROM check_input log</p>
</div>
<div class="section" id="what-is-the-makefile">
<h2>What is the Makefile?<a class="headerlink" href="#what-is-the-makefile" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="what-is-the-config-txt-file">
<h2>What is the config.txt file?<a class="headerlink" href="#what-is-the-config-txt-file" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="how-to-regenerate-a-new-database-version">
<h2>How to regenerate a new database version?<a class="headerlink" href="#how-to-regenerate-a-new-database-version" title="Permalink to this headline">¶</a></h2>
<p>Voir les notes de Jeanne sur le problème de Sebastian</p>
</div>
<div class="section" id="how-to-map-a-metabolic-network-on-another-database">
<span id="map-database"></span><h2>How to map a metabolic network on another database?<a class="headerlink" href="#how-to-map-a-metabolic-network-on-another-database" title="Permalink to this headline">¶</a></h2>
<p>Metabolic networks can be products of varied databases. If you want to
merge efficiently information about metabolic networks coming from
different databases, you will need to map the metabolic network(s) to a
common database. To do so, a solution is provided be AuReMe.</p>
<p>Note: to use this method, the metabolic network to map needs to be in
the SBML format and stored in the <strong>*networks*</strong> folder.</p>
<ul>
<li><div class="first line-block">
<div class="line">First of all, you need to know the origin database of the data. To
recognize the database used in an SBML file, use the <strong>*which_db*</strong>
command:</div>
<div class="line">Example:</div>
</div>
<p><a href="#id3"><span class="problematic" id="id4">**</span></a>[output] **</p>
</li>
<li><p class="first">When you know the origin database of the data, you have to generate
the mapping dictionary from this database to the new one:</p>
<p>Example:</p>
<p><a href="#id5"><span class="problematic" id="id6">**</span></a>[output] **</p>
<p>In this example, the system has found more than just one mapping for
the <em>R_R00494_c</em> reaction and the <em>S_Starch_p</em> compound. It did not
manage to choose between the propositions: the mapping will not be
added to the output mapping. If you want to force the mapping, you
have to modify the mapping file manually.</p>
</li>
<li><p class="first">Once you have created a mapping dictionary file, it will be
automatically applied across the workflow to translate the data.</p>
</li>
</ul>
</div>
<div class="section" id="how-to-generate-reports-on-results">
<h2>How to generate reports on results?<a class="headerlink" href="#how-to-generate-reports-on-results" title="Permalink to this headline">¶</a></h2>
<p>Create reports on the <em>network_name</em> network (in the <strong>*networks*</strong>
directory). The reports is created in the <strong>*analysisreports*</strong>
directory.</p>
<p>Crée 4 fichiers bridge/test/analysis/report/network_name:</p>
<ul class="simple">
<li>All_genes :</li>
</ul>
<p>Id common name linked reactions (;)</p>
<ul class="simple">
<li>All_metabolites</li>
</ul>
<p>dbRef_id common name Produced (p), Consumed (c), Both (cp)</p>
<ul class="simple">
<li>All_pathways</li>
</ul>
<p>dbRef_id common name Number of reaction found Total number of reaction
Ratio</p>
<ul class="simple">
<li>All_reactions</li>
</ul>
<p>nbRef_id common name formula (with ID) formula (with common name) in
pathway associated genes categories</p>
</div>
<div class="section" id="how-to-generate-wiki">
<h2>How to generate Wiki?<a class="headerlink" href="#how-to-generate-wiki" title="Permalink to this headline">¶</a></h2>
<p>Voir la formation de Méziane</p>
<p><a class="reference internal" href="media/image10.png"><img alt="image9" src="media/image10.png" style="width: 1.35383in; height: 2.16535in;" /></a></p>
<ol class="arabic simple">
<li>Create a wiki<ol class="loweralpha">
<li>Create the wiki pages. The pages will be in
analysis/wiki_pages/network_name</li>
</ol>
</li>
</ol>
<p>Wiki_Docker is an image that allows to automatize the creation of wiki
in containers.</p>
<ul class="simple">
<li>Run the next commands from your machine and not from the AuReMe
container.<ol class="loweralpha" start="2">
<li>Download the wiki docker image.</li>
<li>Run and setup a container with wiki docker. Follow the
instructions to setup correctly the wiki.</li>
<li>Send the pages and the configuration to the wiki</li>
</ol>
</li>
</ul>
</div>
<div class="section" id="how-to-connect-to-pathway-tools">
<h2>How to connect to Pathway-tools?<a class="headerlink" href="#how-to-connect-to-pathway-tools" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Create PGDB from output of AuReMe</li>
</ul>
</div>
<div class="section" id="what-are-artefacts">
<span id="artefacts"></span><h2>What are “artefacts”?<a class="headerlink" href="#what-are-artefacts" title="Permalink to this headline">¶</a></h2>
<p>Meneco is a tool that fill the gaps topologically in a network, thanks to a
reference database (see the <a class="reference internal" href="aureme.html#meneco"><span class="std std-ref">Method: Meneco</span></a> section). In fact, Meneco cannot
product any other metabolite of an cycle without initiate it before.
Thereby, artefacts are metabolites allow Meneco to initiate cycles in a
metabolic network.
For example in the picture aside, the Kreps cycle needs to be initiated with
Meneco. A manner to initiate the Kreps cycle into Meneco is to put the
“citrate” metabolite as one of the “artefacts” before gap-filling the
network thanks to Meneco.</p>
</div>
<div class="section" id="how-to-process-flux-balance-analysis">
<h2>How to process Flux Balance Analysis?<a class="headerlink" href="#how-to-process-flux-balance-analysis" title="Permalink to this headline">¶</a></h2>
<p>Notes Mez</p>
<p>To set the objective reaction, please see the following FAQ section.</p>
</div>
<div class="section" id="how-to-set-an-objective-reaction">
<h2>How to set an objective reaction?<a class="headerlink" href="#how-to-set-an-objective-reaction" title="Permalink to this headline">¶</a></h2>
<p>Notes Mez</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="aureme.html" class="btn btn-neutral" title="How to use the AuReMe workspace (default workflow)" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Meziane AITE

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    
    
      <script type="text/javascript">
          var DOCUMENTATION_OPTIONS = {
              URL_ROOT:'./',
              VERSION:'',
              LANGUAGE:'None',
              COLLAPSE_INDEX:false,
              FILE_SUFFIX:'.html',
              HAS_SOURCE:  true,
              SOURCELINK_SUFFIX: '.txt'
          };
      </script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    

  

  <script type="text/javascript" src="_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>